<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>EigenGWAS theory and application</title>
  <meta name="description" content="EigenGWAS Project.">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="EigenGWAS theory and application" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="EigenGWAS Project." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="EigenGWAS theory and application" />
  
  <meta name="twitter:description" content="EigenGWAS Project." />
  

<meta name="author" content="Guo-Bo Chen [chen.guobo@foxmail.com]">


<meta name="date" content="2019-10-28">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="IBD-fam.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EigenGWAS core</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> EigenGWAS basis</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#genetic-relatedness-matrix-mathbfg"><i class="fa fa-check"></i><b>1.1</b> Genetic relatedness matrix <span class="math inline">\(\mathbf{G}\)</span></a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#statistical-properties-of-mathbfg"><i class="fa fa-check"></i><b>1.1.1</b> Statistical properties of <span class="math inline">\(\mathbf{G}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#eigengwas-linear-model"><i class="fa fa-check"></i><b>1.2</b> EigenGWAS linear model</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#lambda_gc-correction"><i class="fa fa-check"></i><b>1.2.1</b> <span class="math inline">\(\lambda_{GC}\)</span> correction</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#threshold-for-eigengwas"><i class="fa fa-check"></i><b>1.2.2</b> Threshold for EigenGWAS</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#connection-to-singular-value-decomposition"><i class="fa fa-check"></i><b>1.3</b> Connection to singular value decomposition</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#intepretation"><i class="fa fa-check"></i><b>1.4</b> Intepretation</a><ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#f_st"><i class="fa fa-check"></i><b>1.4.1</b> <span class="math inline">\(F_{st}\)</span></a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#classic-mechanic-intepretation"><i class="fa fa-check"></i><b>1.4.2</b> Classic mechanic intepretation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="protocols.html"><a href="protocols.html"><i class="fa fa-check"></i><b>2</b> Protocols</a><ul>
<li class="chapter" data-level="2.1" data-path="protocols.html"><a href="protocols.html#protocols-for-selection"><i class="fa fa-check"></i><b>2.1</b> Protocols for selection</a></li>
<li class="chapter" data-level="2.2" data-path="protocols.html"><a href="protocols.html#protocal-for-eigengwas"><i class="fa fa-check"></i><b>2.2</b> Protocal for EigenGWAS</a></li>
<li class="chapter" data-level="2.3" data-path="protocols.html"><a href="protocols.html#protocol-for-predicted-eigenvectors"><i class="fa fa-check"></i><b>2.3</b> Protocol for predicted eigenvectors</a></li>
<li class="chapter" data-level="2.4" data-path="protocols.html"><a href="protocols.html#nucleotide-diversity"><i class="fa fa-check"></i><b>2.4</b> Nucleotide diversity</a></li>
<li class="chapter" data-level="2.5" data-path="protocols.html"><a href="protocols.html#other-test-statistics-for-selection"><i class="fa fa-check"></i><b>2.5</b> Other test statistics for selection</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="resequencing-studies.html"><a href="resequencing-studies.html"><i class="fa fa-check"></i><b>3</b> Resequencing studies</a><ul>
<li class="chapter" data-level="3.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#technical-review"><i class="fa fa-check"></i><b>3.1</b> Technical review</a><ul>
<li class="chapter" data-level="3.1.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#simulation"><i class="fa fa-check"></i><b>3.1.1</b> Simulation</a></li>
<li class="chapter" data-level="3.1.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#ngs"><i class="fa fa-check"></i><b>3.1.2</b> NGS</a></li>
<li class="chapter" data-level="3.1.3" data-path="resequencing-studies.html"><a href="resequencing-studies.html#chip-data"><i class="fa fa-check"></i><b>3.1.3</b> Chip data</a></li>
<li class="chapter" data-level="3.1.4" data-path="resequencing-studies.html"><a href="resequencing-studies.html#gbs"><i class="fa fa-check"></i><b>3.1.4</b> GBS</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#drop-hwe-test"><i class="fa fa-check"></i><b>3.2</b> Drop HWE test</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html"><i class="fa fa-check"></i><b>4</b> Simulating population structure</a><ul>
<li class="chapter" data-level="4.1" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#genetic-drift"><i class="fa fa-check"></i><b>4.1</b> Genetic drift</a></li>
<li class="chapter" data-level="4.2" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#discrete-populations"><i class="fa fa-check"></i><b>4.2</b> Discrete populations</a></li>
<li class="chapter" data-level="4.3" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#admixture-populations"><i class="fa fa-check"></i><b>4.3</b> Admixture populations</a></li>
<li class="chapter" data-level="4.4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#homo-heteogeneous-f_st"><i class="fa fa-check"></i><b>4.4</b> Homo &amp; Heteogeneous <span class="math inline">\(F_{st}\)</span></a></li>
<li class="chapter" data-level="4.5" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#wishart-distribution"><i class="fa fa-check"></i><b>4.5</b> Wishart distribution</a></li>
<li class="chapter" data-level="4.6" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#tracy-widom-distribution"><i class="fa fa-check"></i><b>4.6</b> Tracy-Widom distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>5</b> Data analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="data-analysis.html"><a href="data-analysis.html#on-site-examples"><i class="fa fa-check"></i><b>5.1</b> On-site examples</a><ul>
<li class="chapter" data-level="5.1.1" data-path="data-analysis.html"><a href="data-analysis.html#arabdiopsis"><i class="fa fa-check"></i><b>5.1.1</b> Arabdiopsis</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-analysis.html"><a href="data-analysis.html#k-rice"><i class="fa fa-check"></i><b>5.1.2</b> 3K rice</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-analysis.html"><a href="data-analysis.html#als"><i class="fa fa-check"></i><b>5.1.3</b> ALS</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-analysis.html"><a href="data-analysis.html#gf"><i class="fa fa-check"></i><b>5.1.4</b> GF</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-analysis.html"><a href="data-analysis.html#uk-birds"><i class="fa fa-check"></i><b>5.1.5</b> UK birds</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-analysis.html"><a href="data-analysis.html#darwins-finches"><i class="fa fa-check"></i><b>5.1.6</b> Darwinâ€™s finches</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-analysis.html"><a href="data-analysis.html#gbs-maize"><i class="fa fa-check"></i><b>5.1.7</b> GBS Maize</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-analysis.html"><a href="data-analysis.html#uk-biobank"><i class="fa fa-check"></i><b>5.1.8</b> UK Biobank</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-analysis.html"><a href="data-analysis.html#public-datahub-neo"><i class="fa fa-check"></i><b>5.2</b> Public datahub (NEO)</a></li>
<li class="chapter" data-level="5.3" data-path="data-analysis.html"><a href="data-analysis.html#meta-scale"><i class="fa fa-check"></i><b>5.3</b> Meta-scale</a><ul>
<li class="chapter" data-level="5.3.1" data-path="data-analysis.html"><a href="data-analysis.html#n_e"><i class="fa fa-check"></i><b>5.3.1</b> <span class="math inline">\(n_e\)</span></a></li>
<li class="chapter" data-level="5.3.2" data-path="data-analysis.html"><a href="data-analysis.html#m_e"><i class="fa fa-check"></i><b>5.3.2</b> <span class="math inline">\(m_e\)</span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a><ul>
<li class="chapter" data-level="6.1" data-path="conclusion.html"><a href="conclusion.html#statistical-power"><i class="fa fa-check"></i><b>6.1</b> Statistical power</a></li>
<li class="chapter" data-level="6.2" data-path="conclusion.html"><a href="conclusion.html#selection-pattern"><i class="fa fa-check"></i><b>6.2</b> Selection pattern</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="app-1.html"><a href="app-1.html"><i class="fa fa-check"></i><b>7</b> Appendix 1</a><ul>
<li class="chapter" data-level="7.1" data-path="app-1.html"><a href="app-1.html#app:ld"><i class="fa fa-check"></i><b>7.1</b> Notes on linkage disequilibrium</a></li>
<li class="chapter" data-level="7.2" data-path="app-1.html"><a href="app-1.html#the-maximal-value-rho_ab"><i class="fa fa-check"></i><b>7.2</b> The maximal value <span class="math inline">\(\rho_{AB}\)</span></a></li>
<li class="chapter" data-level="7.3" data-path="app-1.html"><a href="app-1.html#ld-and-recombination-fraction"><i class="fa fa-check"></i><b>7.3</b> LD and recombination fraction</a></li>
<li class="chapter" data-level="7.4" data-path="app-1.html"><a href="app-1.html#f_st-1"><i class="fa fa-check"></i><b>7.4</b> <span class="math inline">\(F_{st}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="8" data-path="notes-1.html"><a href="notes-1.html"><i class="fa fa-check"></i><b>8</b> Notes</a><ul>
<li class="chapter" data-level="8.1" data-path="notes-1.html"><a href="notes-1.html#apy"><i class="fa fa-check"></i><b>8.1</b> APY</a><ul>
<li class="chapter" data-level="8.1.1" data-path="notes-1.html"><a href="notes-1.html#a-numerical-example"><i class="fa fa-check"></i><b>8.1.1</b> A numerical example</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="notes-1.html"><a href="notes-1.html#ld-score-regression"><i class="fa fa-check"></i><b>8.2</b> LD score regression</a></li>
<li class="chapter" data-level="8.3" data-path="notes-1.html"><a href="notes-1.html#matrix-inversion-wiki"><i class="fa fa-check"></i><b>8.3</b> Matrix inversion <span>wiki</span></a><ul>
<li class="chapter" data-level="8.3.1" data-path="notes-1.html"><a href="notes-1.html#x-2"><i class="fa fa-check"></i><b>8.3.1</b> 2 X 2</a></li>
<li class="chapter" data-level="8.3.2" data-path="notes-1.html"><a href="notes-1.html#x-3"><i class="fa fa-check"></i><b>8.3.2</b> 3 X 3</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="IBD-fam.html"><a href="IBD-fam.html"><i class="fa fa-check"></i><b>9</b> IBD notes</a><ul>
<li class="chapter" data-level="9.1" data-path="IBD-fam.html"><a href="IBD-fam.html#he-association"><i class="fa fa-check"></i><b>9.1</b> HE association</a></li>
<li class="chapter" data-level="9.2" data-path="IBD-fam.html"><a href="IBD-fam.html#ibd-table"><i class="fa fa-check"></i><b>9.2</b> IBD table</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="app-2.html"><a href="app-2.html"><i class="fa fa-check"></i><b>10</b> Appendix quick pca</a></li>
<li class="divider"></li>
<li><a href="https://gc5k.github.io" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EigenGWAS theory and application</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="app-2" class="section level1">
<h1><span class="header-section-number">Chapter 10</span> Appendix quick pca</h1>
<p>See PLoS ONE, 2014, 9:e93766</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(MASS)</code></pre>
<pre><code>## Warning: package &#39;MASS&#39; was built under R version 3.4.4</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">N=<span class="dv">1000</span> <span class="co">#sample size</span>
M=<span class="dv">2000</span> <span class="co">#SNP</span>
X=<span class="kw">matrix</span>(<span class="dv">0</span>, N, M) <span class="co">#SNP matrix</span>

<span class="co">#simulating snp</span>
<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>M) {
  p1=<span class="kw">runif</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="fl">0.9</span>)
  p2=<span class="dv">1</span><span class="op">-</span>p1
  X[<span class="dv">1</span><span class="op">:</span>(N<span class="op">/</span><span class="dv">2</span>),i]=<span class="kw">rbinom</span>(N<span class="op">/</span><span class="dv">2</span>, <span class="dv">2</span>, p1)
  X[(N<span class="op">/</span><span class="dv">2</span><span class="op">+</span><span class="dv">1</span>)<span class="op">:</span>N,i]=<span class="kw">rbinom</span>(N<span class="op">/</span><span class="dv">2</span>, <span class="dv">2</span>, p2)
}

<span class="co">#conventional PCA</span>
sX=<span class="kw">scale</span>(X)
G=sX<span class="op">%*%</span><span class="kw">t</span>(sX)
Geg=<span class="kw">eigen</span>(G)
<span class="co">#plot(Geg$vectors[,1], Geg$vectors[,2])</span>


<span class="co">#quick pca</span>
dm=<span class="dv">30</span>
sg=<span class="kw">matrix</span>(<span class="dv">0</span>,dm,dm)
<span class="kw">diag</span>(sg)=<span class="dv">1</span>
R=<span class="kw">mvrnorm</span>(M, <span class="kw">rep</span>(<span class="dv">0</span>, dm), <span class="dt">Sigma=</span>sg)
xt=sX<span class="op">%*%</span>R
ss=<span class="kw">apply</span>(xt<span class="op">^</span><span class="dv">2</span>, <span class="dv">2</span>, sum)
Y=<span class="kw">matrix</span>(<span class="dv">0</span>, <span class="kw">nrow</span>(xt), <span class="kw">ncol</span>(xt))
<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(ss)) {
  Y[,i]=xt[,i]<span class="op">/</span>ss[i]
}

XtX=sX<span class="op">%*%</span><span class="kw">t</span>(sX)

maxiter=<span class="dv">10</span>
<span class="cf">for</span>(it <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>maxiter) {
  xxt=XtX<span class="op">%*%</span>Y
  ss1=<span class="kw">apply</span>(xxt<span class="op">^</span><span class="dv">2</span>, <span class="dv">2</span>, sum)
  
  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(ss1)) {
    Y[,i]=xxt[,i]<span class="op">/</span>ss1[i]
  }
}

QR=<span class="kw">qr.default</span>(Y)
B=<span class="kw">t</span>(QR<span class="op">$</span>qr)<span class="op">%*%</span>sX
S=B<span class="op">%*%</span><span class="kw">t</span>(B)

eg=<span class="kw">eigen</span>(S)

U=QR<span class="op">$</span>qr <span class="op">%*%</span><span class="st"> </span>eg<span class="op">$</span>vectors
D=<span class="kw">sqrt</span>(eg<span class="op">$</span>values<span class="op">/</span>(N<span class="dv">-1</span>))  
P=U<span class="op">*</span>D

<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">plot</span>(<span class="dt">main=</span><span class="st">&quot;PCA&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;eVec 1&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;eVec 2&quot;</span>, Geg<span class="op">$</span>vectors[,<span class="dv">1</span>], Geg<span class="op">$</span>vectors[,<span class="dv">2</span>])
<span class="kw">plot</span>(<span class="dt">main=</span><span class="st">&quot;Quick PCA&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;eVec 1&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;eVec 2&quot;</span>, U[,<span class="dv">1</span>], U[,<span class="dv">2</span>])</code></pre>
<p><img src="EigenGWAS_files/figure-html/quick-pca-1.png" width="672" /></p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">cor</span>(Geg<span class="op">$</span>vectors[,<span class="dv">1</span>], U[,<span class="dv">1</span>])</code></pre>
<pre><code>## [1] 0.991082</code></pre>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="IBD-fam.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["EigenGWAS.pdf", "EigenGWAS.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
