<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>EigenGWAS theory and application</title>
  <meta name="description" content="EigenGWAS Project.">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="EigenGWAS theory and application" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="EigenGWAS Project." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="EigenGWAS theory and application" />
  
  <meta name="twitter:description" content="EigenGWAS Project." />
  

<meta name="author" content="Guo-Bo Chen [chen.guobo@foxmail.com]">


<meta name="date" content="2018-11-29">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="conclusion.html">
<link rel="next" href="references.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EigenGWAS core</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> EigenGWAS basis</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#genetic-relatedness-matrix-mathbfg"><i class="fa fa-check"></i><b>1.1</b> Genetic relatedness matrix <span class="math inline">\(\mathbf{G}\)</span></a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#statistical-properties-of-mathbfg"><i class="fa fa-check"></i><b>1.1.1</b> Statistical properties of <span class="math inline">\(\mathbf{G}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#eigengwas-linear-model"><i class="fa fa-check"></i><b>1.2</b> EigenGWAS linear model</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#lambda_gc-correction"><i class="fa fa-check"></i><b>1.2.1</b> <span class="math inline">\(\lambda_{GC}\)</span> correction</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#connection-to-svd"><i class="fa fa-check"></i><b>1.3</b> Connection to SVD</a><ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#threshold-for-eigengwas"><i class="fa fa-check"></i><b>1.3.1</b> Threshold for EigenGWAS</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#classic-mechanic-intepretation"><i class="fa fa-check"></i><b>1.4</b> Classic mechanic intepretation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="protocols.html"><a href="protocols.html"><i class="fa fa-check"></i><b>2</b> Protocols</a><ul>
<li class="chapter" data-level="2.1" data-path="protocols.html"><a href="protocols.html#protocols-for-selection"><i class="fa fa-check"></i><b>2.1</b> Protocols for selection</a></li>
<li class="chapter" data-level="2.2" data-path="protocols.html"><a href="protocols.html#protocal-for-eigengwas"><i class="fa fa-check"></i><b>2.2</b> Protocal for EigenGWAS</a></li>
<li class="chapter" data-level="2.3" data-path="protocols.html"><a href="protocols.html#protocol-for-predicted-eigenvectors"><i class="fa fa-check"></i><b>2.3</b> Protocol for predicted eigenvectors</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="resequencing-studies.html"><a href="resequencing-studies.html"><i class="fa fa-check"></i><b>3</b> Resequencing studies</a><ul>
<li class="chapter" data-level="3.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#technical-review"><i class="fa fa-check"></i><b>3.1</b> Technical review</a><ul>
<li class="chapter" data-level="3.1.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#ngs"><i class="fa fa-check"></i><b>3.1.1</b> NGS</a></li>
<li class="chapter" data-level="3.1.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#chip-data"><i class="fa fa-check"></i><b>3.1.2</b> Chip data</a></li>
<li class="chapter" data-level="3.1.3" data-path="resequencing-studies.html"><a href="resequencing-studies.html#gbs"><i class="fa fa-check"></i><b>3.1.3</b> GBS</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#drop-hwe-test"><i class="fa fa-check"></i><b>3.2</b> Drop HWE test</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html"><i class="fa fa-check"></i><b>4</b> Simulating population structure</a><ul>
<li class="chapter" data-level="4.1" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#genetic-drift"><i class="fa fa-check"></i><b>4.1</b> Genetic drift</a></li>
<li class="chapter" data-level="4.2" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#three-pop"><i class="fa fa-check"></i><b>4.2</b> Three pop</a></li>
<li class="chapter" data-level="4.3" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#six-pop"><i class="fa fa-check"></i><b>4.3</b> Six pop</a></li>
<li class="chapter" data-level="4.4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#wishart-distribution-simulation"><i class="fa fa-check"></i><b>4.4</b> Wishart distribution simulation</a></li>
<li class="chapter" data-level="4.5" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#distribution-of-the-wishart-diagonal-elements"><i class="fa fa-check"></i><b>4.5</b> Distribution of the Wishart diagonal elements</a></li>
<li class="chapter" data-level="4.6" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#homo-cohort"><i class="fa fa-check"></i><b>4.6</b> Homo cohort</a></li>
<li class="chapter" data-level="4.7" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#heterogeneous-cohort"><i class="fa fa-check"></i><b>4.7</b> Heterogeneous cohort</a></li>
<li class="chapter" data-level="4.8" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#tracy-widom-distribution"><i class="fa fa-check"></i><b>4.8</b> Tracy-Widom distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>5</b> Applications</a><ul>
<li class="chapter" data-level="5.1" data-path="applications.html"><a href="applications.html#analysis-pipeline"><i class="fa fa-check"></i><b>5.1</b> Analysis pipeline</a><ul>
<li class="chapter" data-level="5.1.1" data-path="applications.html"><a href="applications.html#commandline-protocol"><i class="fa fa-check"></i><b>5.1.1</b> Commandline protocol</a></li>
<li class="chapter" data-level="5.1.2" data-path="applications.html"><a href="applications.html#web-interactive-tools"><i class="fa fa-check"></i><b>5.1.2</b> Web interactive tools</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="applications.html"><a href="applications.html#examples"><i class="fa fa-check"></i><b>5.2</b> Examples</a><ul>
<li class="chapter" data-level="5.2.1" data-path="applications.html"><a href="applications.html#arabdiopsis"><i class="fa fa-check"></i><b>5.2.1</b> Arabdiopsis</a></li>
<li class="chapter" data-level="5.2.2" data-path="applications.html"><a href="applications.html#als"><i class="fa fa-check"></i><b>5.2.2</b> ALS</a></li>
<li class="chapter" data-level="5.2.3" data-path="applications.html"><a href="applications.html#gf"><i class="fa fa-check"></i><b>5.2.3</b> GF</a></li>
<li class="chapter" data-level="5.2.4" data-path="applications.html"><a href="applications.html#uk-birds"><i class="fa fa-check"></i><b>5.2.4</b> UK birds</a></li>
<li class="chapter" data-level="5.2.5" data-path="applications.html"><a href="applications.html#darwins-finches"><i class="fa fa-check"></i><b>5.2.5</b> Darwin’s finches</a></li>
<li class="chapter" data-level="5.2.6" data-path="applications.html"><a href="applications.html#maize-li-hh"><i class="fa fa-check"></i><b>5.2.6</b> Maize (Li HH)</a></li>
<li class="chapter" data-level="5.2.7" data-path="applications.html"><a href="applications.html#k-rice"><i class="fa fa-check"></i><b>5.2.7</b> 3K rice</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="applications.html"><a href="applications.html#sea-volumn-data-neo"><i class="fa fa-check"></i><b>5.3</b> Sea volumn data (NEO)</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>6</b> Data analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="data-analysis.html"><a href="data-analysis.html#on-site-examples"><i class="fa fa-check"></i><b>6.1</b> On site examples</a><ul>
<li class="chapter" data-level="6.1.1" data-path="data-analysis.html"><a href="data-analysis.html#arabdiopsis-1"><i class="fa fa-check"></i><b>6.1.1</b> Arabdiopsis</a></li>
<li class="chapter" data-level="6.1.2" data-path="data-analysis.html"><a href="data-analysis.html#k-rice-1"><i class="fa fa-check"></i><b>6.1.2</b> 3K rice</a></li>
<li class="chapter" data-level="6.1.3" data-path="data-analysis.html"><a href="data-analysis.html#als-1"><i class="fa fa-check"></i><b>6.1.3</b> ALS</a></li>
<li class="chapter" data-level="6.1.4" data-path="data-analysis.html"><a href="data-analysis.html#gf-1"><i class="fa fa-check"></i><b>6.1.4</b> GF</a></li>
<li class="chapter" data-level="6.1.5" data-path="data-analysis.html"><a href="data-analysis.html#uk-birds-1"><i class="fa fa-check"></i><b>6.1.5</b> UK birds</a></li>
<li class="chapter" data-level="6.1.6" data-path="data-analysis.html"><a href="data-analysis.html#darwins-finches-1"><i class="fa fa-check"></i><b>6.1.6</b> Darwin’s finches</a></li>
<li class="chapter" data-level="6.1.7" data-path="data-analysis.html"><a href="data-analysis.html#gbs-maize"><i class="fa fa-check"></i><b>6.1.7</b> GBS Maize</a></li>
<li class="chapter" data-level="6.1.8" data-path="data-analysis.html"><a href="data-analysis.html#uk-biobank"><i class="fa fa-check"></i><b>6.1.8</b> UK Biobank</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="data-analysis.html"><a href="data-analysis.html#public-datahub-neo"><i class="fa fa-check"></i><b>6.2</b> Public datahub (NEO)</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="meta-research.html"><a href="meta-research.html"><i class="fa fa-check"></i><b>7</b> Meta-research</a><ul>
<li class="chapter" data-level="7.1" data-path="meta-research.html"><a href="meta-research.html#n_e"><i class="fa fa-check"></i><b>7.1</b> <span class="math inline">\(n_e\)</span></a></li>
<li class="chapter" data-level="7.2" data-path="meta-research.html"><a href="meta-research.html#m_e"><i class="fa fa-check"></i><b>7.2</b> <span class="math inline">\(m_e\)</span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>8</b> Conclusion</a><ul>
<li class="chapter" data-level="8.1" data-path="conclusion.html"><a href="conclusion.html#statistical-power"><i class="fa fa-check"></i><b>8.1</b> Statistical power</a></li>
<li class="chapter" data-level="8.2" data-path="conclusion.html"><a href="conclusion.html#selection-pattern"><i class="fa fa-check"></i><b>8.2</b> Selection pattern</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="app-1.html"><a href="app-1.html"><i class="fa fa-check"></i><b>9</b> Appendix 1</a><ul>
<li class="chapter" data-level="9.1" data-path="app-1.html"><a href="app-1.html#app:ld"><i class="fa fa-check"></i><b>9.1</b> Notes on linkage disequilibrium</a></li>
<li class="chapter" data-level="9.2" data-path="app-1.html"><a href="app-1.html#the-maximal-value-rho_ab"><i class="fa fa-check"></i><b>9.2</b> The maximal value <span class="math inline">\(\rho_{AB}\)</span></a></li>
<li class="chapter" data-level="9.3" data-path="app-1.html"><a href="app-1.html#ld-and-recombination-fraction"><i class="fa fa-check"></i><b>9.3</b> LD and recombination fraction</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://gc5k.github.io" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EigenGWAS theory and application</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="app-1" class="section level1">
<h1><span class="header-section-number">Chapter 9</span> Appendix 1</h1>
<div id="app:ld" class="section level2">
<h2><span class="header-section-number">9.1</span> Notes on linkage disequilibrium</h2>
<p>For the haplotypes, as tabulated below, consisting of locus 1, which has alleles <span class="math inline">\(A\)</span> and <span class="math inline">\(a\)</span>, and locus 2, which has alleles <span class="math inline">\(B\)</span> and <span class="math inline">\(b\)</span>,</p>
<table>
<thead>
<tr class="header">
<th align="center"></th>
<th align="center"><span class="math inline">\(B\)</span></th>
<th align="center"><span class="math inline">\(b\)</span></th>
<th align="center"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><span class="math inline">\(A\)</span></td>
<td align="center"><span class="math inline">\(P_{AB}=P_A \times P_B + D\)</span></td>
<td align="center"><span class="math inline">\(P_{Ab}=P_A \times P_b - D\)</span></td>
<td align="center"><span class="math inline">\(P_A\)</span></td>
</tr>
<tr class="even">
<td align="center"><span class="math inline">\(a\)</span></td>
<td align="center"><span class="math inline">\(P_{aB}=P_a \times P_B - D\)</span></td>
<td align="center"><span class="math inline">\(P_{ab}=P_a \times P_b + D\)</span></td>
<td align="center"><span class="math inline">\(P_a\)</span></td>
</tr>
<tr class="odd">
<td align="center"></td>
<td align="center"><span class="math inline">\(P_B\)</span></td>
<td align="center"><span class="math inline">\(P_b\)</span></td>
<td align="center"></td>
</tr>
</tbody>
</table>
<p>It is a typical <span class="math inline">\(2\times2\)</span> contingency table, once one of these four cell is fixed, other three are all known. Consequently, we can implement Fisher’s Exact test for this table.
However, given the current biotechnology, <span class="math inline">\(D\)</span> is often unknown. But given the correlation, <span class="math inline">\(\rho_{1,2}\)</span>, of a pair of markers, we know that
<span class="math display">\[\rho_{1,2}=\frac{D_{1,2}}{\sqrt{P_AP_aP_BP_b}}\]</span></p>
<p>In simulation, it is very easy to simulate LD given the correlation.</p>
<p>Calculate <span class="math inline">\(D_{1,2}=\rho_{1,2}\sqrt{P_AP_aP_BP_b}\)</span> first.</p>
<p>Depending on the allele at locus 1, the probabilities of observing allele <span class="math inline">\(B\)</span> and <span class="math inline">\(b\)</span> are <span class="math inline">\(P(B|A)=\frac{P_{AB}}{P_{A}} =\frac{P_A\times P_B+D}{P_A}\)</span>, <span class="math inline">\(P(b|A)=1-P(B|A)\)</span>, <span class="math inline">\(P(b|a)=\frac{P_{ab}}{P_a} =\frac{P_a\times P_b+D}{P_a}\)</span>, and <span class="math inline">\(P(B|a)=1-P(B|a)\)</span>, respectively. So, the recombination can be defined as <span class="math inline">\(\frac{P(B|A)}{P(B|A) + P(b|A)}=P_B + \frac{D}{P_A}\)</span>.</p>
</div>
<div id="the-maximal-value-rho_ab" class="section level2">
<h2><span class="header-section-number">9.2</span> The maximal value <span class="math inline">\(\rho_{AB}\)</span></h2>
<p>Assuming <span class="math inline">\(P_A \leq P_B \leq 0.5\)</span>, the maximal of <span class="math inline">\(\rho_{l_1l_2}\)</span> reaches when allele <span class="math inline">\(A\)</span> is <span class="math inline">\(a\)</span> subset of <span class="math inline">\(B\)</span> linage so that <span class="math inline">\(P_{AB}=P_{A}\)</span>, because (see PLoS Genet 2(9): e142)
<span class="math display">\[\rho_{1,2}=\frac{P_{AB}-P_{A}P_{B}}{\sqrt{P_A P_a P_B P_b}}=\frac{P_A(1-P_B)}{\sqrt{P_A P_a P_B P_b}}=\sqrt{\frac{P_AP_b}{P_B P_a}} \leq 1\]</span></p>
</div>
<div id="ld-and-recombination-fraction" class="section level2">
<h2><span class="header-section-number">9.3</span> LD and recombination fraction</h2>
<p>It is sometimes useful to simulate <span class="math inline">\(F_2\)</span> population given known LD between any pair of markers. Below is the procedure for doing that.</p>
<p>For a pair of loci, the correlation is the function of their recombination function
<span class="math display">\[\rho_{1,2}=1-2c_{1,2}\]</span></p>
<p>In addition, the correlation is also the function of LD
<span class="math display">\[\rho_{1,2}=\frac{D_{1,2}}{\sqrt{P_AP_aP_BP_b}}\]</span></p>
<p>So given the allele frequency of 0.5 and the markers are evenly distributed, <span class="math inline">\(c_{1,2}=0.5-2D_{1,2}\)</span>, and their corresponding genetic distance is <span class="math inline">\(d_{1,2}=-\frac{1}{2}ln(1-2c_{1,2})=-\frac{1}{2}ln(4D_{1,2})\)</span>
given <code>Haldane map function</code>.</p>
<p>Under the Haldane map function, the recombination between locus 1 and 3 is <span class="math inline">\(c_{1,3}=\frac{1}{2}[1-e^{-2(d_{1,2}+d_{2,3})}]=\frac{1}{2}[1-e^{2ln(4D_{1,2})}]\)</span>
Then the correlation between the locus 1 and 3 is <span class="math inline">\(\rho_{1,3}=1-2c_{1,3}=e^{2ln(4D_{1,2})}\)</span>
And, their corresponding LD is
<span class="math inline">\(D_{1,3}=\sqrt{P_AP_aP_BP_b}e^{ln(4D_{l_1l_2})}=0.25e^{2ln(4D_{l_1l_2})}\)</span></p>
<p>In general, <span class="math inline">\(D_{l_il_j}=0.25e^{|l_i-l_j|}ln(4D_{l_1l_2})\)</span>.</p>
<pre class="sourceCode r"><code class="sourceCode r">A =<span class="st"> </span><span class="kw">seq</span>(<span class="fl">0.01</span>, <span class="fl">0.5</span>, <span class="fl">0.01</span>)
B =<span class="st"> </span><span class="kw">seq</span>(<span class="fl">0.01</span>, <span class="fl">0.5</span>, <span class="fl">0.01</span>)

C =<span class="st"> </span><span class="fl">0.8</span>
M =<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="kw">length</span>(A), <span class="kw">length</span>(B))
<span class="kw">plot</span>(<span class="dt">x =</span> <span class="ot">NULL</span>, <span class="dt">y =</span> <span class="ot">NULL</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="dt">xlab =</span> <span class="st">&quot;MAF Locus 1&quot;</span>, 
    <span class="dt">ylab =</span> <span class="st">&quot;MAF Locus 2&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;the relationship between D&quot;</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(A)) {
    <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(B)) {
        M[i, j] =<span class="st"> </span>C <span class="op">*</span><span class="st"> </span><span class="kw">sqrt</span>(A[i] <span class="op">*</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>A[i]) <span class="op">*</span><span class="st"> </span>B[j] <span class="op">*</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>B[j]))
        <span class="kw">points</span>(A[i], B[j], <span class="dt">col =</span> <span class="kw">rgb</span>(M[i, j]<span class="op">/</span><span class="fl">0.2</span>, <span class="dt">green =</span> <span class="dv">0</span>, <span class="dt">blue =</span> <span class="dv">0</span>), <span class="dt">pch =</span> <span class="dv">16</span>, 
            <span class="dt">cex =</span> <span class="fl">0.3</span>)
    }
}</code></pre>
<p><img src="EigenGWAS_files/figure-html/ld-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="conclusion.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["EigenGWAS.pdf", "EigenGWAS.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
