<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>EigenGWAS theory and application</title>
  <meta name="description" content="EigenGWAS Project.">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="EigenGWAS theory and application" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="EigenGWAS Project." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="EigenGWAS theory and application" />
  
  <meta name="twitter:description" content="EigenGWAS Project." />
  

<meta name="author" content="Guo-Bo Chen [chen.guobo@foxmail.com]">


<meta name="date" content="2018-12-03">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  

<link rel="next" href="protocols.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EigenGWAS core</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> EigenGWAS basis</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#genetic-relatedness-matrix-mathbfg"><i class="fa fa-check"></i><b>1.1</b> Genetic relatedness matrix <span class="math inline">\(\mathbf{G}\)</span></a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#statistical-properties-of-mathbfg"><i class="fa fa-check"></i><b>1.1.1</b> Statistical properties of <span class="math inline">\(\mathbf{G}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#eigengwas-linear-model"><i class="fa fa-check"></i><b>1.2</b> EigenGWAS linear model</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#lambda_gc-correction"><i class="fa fa-check"></i><b>1.2.1</b> <span class="math inline">\(\lambda_{GC}\)</span> correction</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#threshold-for-eigengwas"><i class="fa fa-check"></i><b>1.2.2</b> Threshold for EigenGWAS</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#connection-to-singular-value-decomposition"><i class="fa fa-check"></i><b>1.3</b> Connection to singular value decomposition</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#intepretation"><i class="fa fa-check"></i><b>1.4</b> Intepretation</a><ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#f_st"><i class="fa fa-check"></i><b>1.4.1</b> <span class="math inline">\(F_{st}\)</span></a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#classic-mechanic-intepretation"><i class="fa fa-check"></i><b>1.4.2</b> Classic mechanic intepretation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="protocols.html"><a href="protocols.html"><i class="fa fa-check"></i><b>2</b> Protocols</a><ul>
<li class="chapter" data-level="2.1" data-path="protocols.html"><a href="protocols.html#protocols-for-selection"><i class="fa fa-check"></i><b>2.1</b> Protocols for selection</a></li>
<li class="chapter" data-level="2.2" data-path="protocols.html"><a href="protocols.html#protocal-for-eigengwas"><i class="fa fa-check"></i><b>2.2</b> Protocal for EigenGWAS</a></li>
<li class="chapter" data-level="2.3" data-path="protocols.html"><a href="protocols.html#protocol-for-predicted-eigenvectors"><i class="fa fa-check"></i><b>2.3</b> Protocol for predicted eigenvectors</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="resequencing-studies.html"><a href="resequencing-studies.html"><i class="fa fa-check"></i><b>3</b> Resequencing studies</a><ul>
<li class="chapter" data-level="3.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#technical-review"><i class="fa fa-check"></i><b>3.1</b> Technical review</a><ul>
<li class="chapter" data-level="3.1.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#ngs"><i class="fa fa-check"></i><b>3.1.1</b> NGS</a></li>
<li class="chapter" data-level="3.1.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#chip-data"><i class="fa fa-check"></i><b>3.1.2</b> Chip data</a></li>
<li class="chapter" data-level="3.1.3" data-path="resequencing-studies.html"><a href="resequencing-studies.html#gbs"><i class="fa fa-check"></i><b>3.1.3</b> GBS</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#drop-hwe-test"><i class="fa fa-check"></i><b>3.2</b> Drop HWE test</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html"><i class="fa fa-check"></i><b>4</b> Simulating population structure</a><ul>
<li class="chapter" data-level="4.1" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#genetic-drift"><i class="fa fa-check"></i><b>4.1</b> Genetic drift</a></li>
<li class="chapter" data-level="4.2" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#discrete-populations"><i class="fa fa-check"></i><b>4.2</b> Discrete populations</a></li>
<li class="chapter" data-level="4.3" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#admixture-population"><i class="fa fa-check"></i><b>4.3</b> Admixture population</a></li>
<li class="chapter" data-level="4.4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#wishart-distribution-simulation"><i class="fa fa-check"></i><b>4.4</b> Wishart distribution simulation</a></li>
<li class="chapter" data-level="4.5" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#distribution-of-the-wishart-diagonal-elements"><i class="fa fa-check"></i><b>4.5</b> Distribution of the Wishart diagonal elements</a></li>
<li class="chapter" data-level="4.6" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#homo-cohort"><i class="fa fa-check"></i><b>4.6</b> Homo cohort</a></li>
<li class="chapter" data-level="4.7" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#heterogeneous-cohort"><i class="fa fa-check"></i><b>4.7</b> Heterogeneous cohort</a></li>
<li class="chapter" data-level="4.8" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#tracy-widom-distribution"><i class="fa fa-check"></i><b>4.8</b> Tracy-Widom distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>5</b> Applications</a><ul>
<li class="chapter" data-level="5.1" data-path="applications.html"><a href="applications.html#analysis-pipeline"><i class="fa fa-check"></i><b>5.1</b> Analysis pipeline</a><ul>
<li class="chapter" data-level="5.1.1" data-path="applications.html"><a href="applications.html#commandline-protocol"><i class="fa fa-check"></i><b>5.1.1</b> Commandline protocol</a></li>
<li class="chapter" data-level="5.1.2" data-path="applications.html"><a href="applications.html#web-interactive-tools"><i class="fa fa-check"></i><b>5.1.2</b> Web interactive tools</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="applications.html"><a href="applications.html#examples"><i class="fa fa-check"></i><b>5.2</b> Examples</a><ul>
<li class="chapter" data-level="5.2.1" data-path="applications.html"><a href="applications.html#arabdiopsis"><i class="fa fa-check"></i><b>5.2.1</b> Arabdiopsis</a></li>
<li class="chapter" data-level="5.2.2" data-path="applications.html"><a href="applications.html#als"><i class="fa fa-check"></i><b>5.2.2</b> ALS</a></li>
<li class="chapter" data-level="5.2.3" data-path="applications.html"><a href="applications.html#gf"><i class="fa fa-check"></i><b>5.2.3</b> GF</a></li>
<li class="chapter" data-level="5.2.4" data-path="applications.html"><a href="applications.html#uk-birds"><i class="fa fa-check"></i><b>5.2.4</b> UK birds</a></li>
<li class="chapter" data-level="5.2.5" data-path="applications.html"><a href="applications.html#darwins-finches"><i class="fa fa-check"></i><b>5.2.5</b> Darwin’s finches</a></li>
<li class="chapter" data-level="5.2.6" data-path="applications.html"><a href="applications.html#maize-li-hh"><i class="fa fa-check"></i><b>5.2.6</b> Maize (Li HH)</a></li>
<li class="chapter" data-level="5.2.7" data-path="applications.html"><a href="applications.html#k-rice"><i class="fa fa-check"></i><b>5.2.7</b> 3K rice</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="applications.html"><a href="applications.html#sea-volumn-data-neo"><i class="fa fa-check"></i><b>5.3</b> Sea volumn data (NEO)</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>6</b> Data analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="data-analysis.html"><a href="data-analysis.html#on-site-examples"><i class="fa fa-check"></i><b>6.1</b> On site examples</a><ul>
<li class="chapter" data-level="6.1.1" data-path="data-analysis.html"><a href="data-analysis.html#arabdiopsis-1"><i class="fa fa-check"></i><b>6.1.1</b> Arabdiopsis</a></li>
<li class="chapter" data-level="6.1.2" data-path="data-analysis.html"><a href="data-analysis.html#k-rice-1"><i class="fa fa-check"></i><b>6.1.2</b> 3K rice</a></li>
<li class="chapter" data-level="6.1.3" data-path="data-analysis.html"><a href="data-analysis.html#als-1"><i class="fa fa-check"></i><b>6.1.3</b> ALS</a></li>
<li class="chapter" data-level="6.1.4" data-path="data-analysis.html"><a href="data-analysis.html#gf-1"><i class="fa fa-check"></i><b>6.1.4</b> GF</a></li>
<li class="chapter" data-level="6.1.5" data-path="data-analysis.html"><a href="data-analysis.html#uk-birds-1"><i class="fa fa-check"></i><b>6.1.5</b> UK birds</a></li>
<li class="chapter" data-level="6.1.6" data-path="data-analysis.html"><a href="data-analysis.html#darwins-finches-1"><i class="fa fa-check"></i><b>6.1.6</b> Darwin’s finches</a></li>
<li class="chapter" data-level="6.1.7" data-path="data-analysis.html"><a href="data-analysis.html#gbs-maize"><i class="fa fa-check"></i><b>6.1.7</b> GBS Maize</a></li>
<li class="chapter" data-level="6.1.8" data-path="data-analysis.html"><a href="data-analysis.html#uk-biobank"><i class="fa fa-check"></i><b>6.1.8</b> UK Biobank</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="data-analysis.html"><a href="data-analysis.html#public-datahub-neo"><i class="fa fa-check"></i><b>6.2</b> Public datahub (NEO)</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="meta-research.html"><a href="meta-research.html"><i class="fa fa-check"></i><b>7</b> Meta-research</a><ul>
<li class="chapter" data-level="7.1" data-path="meta-research.html"><a href="meta-research.html#n_e"><i class="fa fa-check"></i><b>7.1</b> <span class="math inline">\(n_e\)</span></a></li>
<li class="chapter" data-level="7.2" data-path="meta-research.html"><a href="meta-research.html#m_e"><i class="fa fa-check"></i><b>7.2</b> <span class="math inline">\(m_e\)</span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>8</b> Conclusion</a><ul>
<li class="chapter" data-level="8.1" data-path="conclusion.html"><a href="conclusion.html#statistical-power"><i class="fa fa-check"></i><b>8.1</b> Statistical power</a></li>
<li class="chapter" data-level="8.2" data-path="conclusion.html"><a href="conclusion.html#selection-pattern"><i class="fa fa-check"></i><b>8.2</b> Selection pattern</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="app-1.html"><a href="app-1.html"><i class="fa fa-check"></i><b>9</b> Appendix 1</a><ul>
<li class="chapter" data-level="9.1" data-path="app-1.html"><a href="app-1.html#app:ld"><i class="fa fa-check"></i><b>9.1</b> Notes on linkage disequilibrium</a></li>
<li class="chapter" data-level="9.2" data-path="app-1.html"><a href="app-1.html#the-maximal-value-rho_ab"><i class="fa fa-check"></i><b>9.2</b> The maximal value <span class="math inline">\(\rho_{AB}\)</span></a></li>
<li class="chapter" data-level="9.3" data-path="app-1.html"><a href="app-1.html#ld-and-recombination-fraction"><i class="fa fa-check"></i><b>9.3</b> LD and recombination fraction</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://gc5k.github.io" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EigenGWAS theory and application</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">EigenGWAS theory and application</h1>
<p class="author"><em>Guo-Bo Chen [<a href="mailto:chen.guobo@foxmail.com" class="email">chen.guobo@foxmail.com</a>]</em></p>
<p class="date"><em>2018-12-03</em></p>
</div>
<div id="ch:basis" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> EigenGWAS basis</h1>
<p>This project is dedicated to <font color="navy"><code>EigenGWAS</code></font>, a linear model analysis approach for genomic data.</p>
<p><font color="navy"><code>EigenGWAS</code></font> can be seamlessly applied to the steps after <code>STRUCTRURE</code> and <code>principal component analysis</code>. <strong>In a nutshell</strong>, <strong><font color="navy">it regresses the eigenvector agaist each SNP</font></strong>, an analysis that finds the loci under selection. It analysis pipeline may be found at <a href="http://rpubs.com/gc5k/EigenGWASpipeline">pipeline</a>.</p>
<div id="genetic-relatedness-matrix-mathbfg" class="section level2">
<h2><span class="header-section-number">1.1</span> Genetic relatedness matrix <span class="math inline">\(\mathbf{G}\)</span></h2>
<p>The genomic data can be represented as <span class="math inline">\(\mathbf{X}\)</span>, an <span class="math inline">\(n \times m\)</span> matrix representing <span class="math inline">\(n\)</span> individuals and <span class="math inline">\(m\)</span> markers. Without loss of generality, <span class="math inline">\(x_{jl}\)</span> is the genotype code for the <span class="math inline">\(i^{th}\)</span> individual at the <span class="math inline">\(l^{th}\)</span> biallic locus. The data matrix <span class="math inline">\(\mathbf{X}\)</span> can be generated from chips, NGS, or GBS.</p>
<p>In general, given <span class="math inline">\(\mathbf{X}\)</span> we can construct the <span class="math inline">\(n\times n\)</span> genetic relatedness matrix as</p>
<p><span class="math display" id="eq:GM">\[\begin{equation}
\mathbf{G}=\frac{1}{m}\tilde{\mathbf{X}}\tilde{\mathbf{X}}^T \tag{1.1}
\end{equation}\]</span></p>
<p>in which <span class="math inline">\(\tilde{\mathbf{X}}\)</span> is the scaled form of <span class="math inline">\(\mathbf{X}\)</span> (slighlty different from standardization, which will be <span class="math inline">\(\frac{x_l-2p_l}{\sqrt{2(1+F_l)p_lq_l}}\)</span>).
However, upon the mating type of the species, <span class="math inline">\(\mathbf{G}\)</span> should be constructed accordingly. For a random mating population, <span class="math inline">\(x_l\)</span> is scaled as <span class="math inline">\(\tilde{x}_l=\frac{x_l-2p_l}{\sqrt{2p_lq_l}}\)</span>, assuming the current population as the base population, whereas for inbred population, <span class="math inline">\(\tilde{x}_l=\frac{x_l-2p_l}{\sqrt{4p_lq_l}}\)</span>, and <span class="math inline">\(q_l=1-p_l\)</span> the frequency for the alternative allele.</p>
<p>Due to missing data, in practice for individual <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, their pairwise relatedness is</p>
<p><span class="math display" id="eq:G">\[\begin{equation}
G_{ij}=\frac{1}{\tilde{m}}\sum_l^{\tilde{m}}\frac{(x_{il}-2p_l)(x_{jl}-2p_l)}{2(1+F)p_lq_l} \tag{1.2}
\end{equation}\]</span></p>
<p>in which <span class="math inline">\(\tilde{m}\)</span> is the number of loci genotyped at both individal <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, and <span class="math inline">\(F\)</span> the inbreeding coefficient taking the value of 0 for random mating population and 1 for inbred population. When <span class="math inline">\(F\)</span> is set as zero, it is assumed that the current population is the <code>base population</code>.</p>
<div id="statistical-properties-of-mathbfg" class="section level3">
<h3><span class="header-section-number">1.1.1</span> Statistical properties of <span class="math inline">\(\mathbf{G}\)</span></h3>
<p>Let <span class="math inline">\(\mathbf{G}_o\)</span> denote the off diagonal elements of <span class="math inline">\(\mathbf{G}\)</span>, then we can define two population parameters, <span class="math inline">\(n_e\)</span>, the <code>effective population size</code>, and <span class="math inline">\(m_e\)</span>, the <code>effective number of markers</code>.</p>
<p><span class="math display" id="eq:Ne">\[\begin{equation}
n_e=\frac{-1}{mean(\mathbf{G}_o)} \tag{1.3}
\end{equation}\]</span>
<span class="math inline">\(n_e\)</span> reflects true relatedness between any pair of samples;</p>
<p><span class="math display" id="eq:Me1">\[\begin{equation}
m_e=\frac{1}{Var(\mathbf{G}_o)}  \tag{1.4}
\end{equation}\]</span>
The ratio between <span class="math inline">\(\frac{m_e}{m}\)</span> reflects the average linkage disequilibrium between the any pair of markers, and alternatively <span class="math inline">\(m_e\)</span> can be expressed as</p>
<p><span class="math display" id="eq:Me2">\[\begin{equation}
m_e=\frac{\sum_{l_1=1}^m\sum_{l_2=1}^m\rho_{l_1l_2}^2}{m^2}=\bar{\rho}^2 \tag{1.5}
\end{equation}\]</span></p>
<p>in which <span class="math inline">\(\rho_{l_1l_2}\)</span> is Pearson’s correlation between a pair of SNPs, see Appendix <a href="app-1.html#app:ld">9.1</a>. It is an important parameter to describe the evolutionary process of a population of study.</p>
</div>
</div>
<div id="eigengwas-linear-model" class="section level2">
<h2><span class="header-section-number">1.2</span> EigenGWAS linear model</h2>
<p>Given eigenanalysis (<a href="https://en.wikipedia.org/wiki/Eigenvalues_and_eigenvectors">see wikipedia for its more details</a>) of <span class="math inline">\(\mathbf{X}\)</span>, we have <span class="math inline">\(\mathbf{E}\)</span> and <span class="math inline">\(\mathbf{\Lambda}\)</span>, in which <span class="math inline">\(\mathbf{\Lambda}\)</span> is an <span class="math inline">\(n \times n\)</span> diagonal matrix for eigenvalues and <span class="math inline">\(\mathbf{E}\)</span> is an <span class="math inline">\(n \times n\)</span> matrix for the eigenvectors. <span class="math inline">\(\mathbf{E}_k\)</span> is the <span class="math inline">\(k^{th}\)</span> eigenvector associated with the <span class="math inline">\(k^{th}\)</span> largest eigenvalue. Regressing <span class="math inline">\(\mathbf{E}_k\)</span> against the <span class="math inline">\(l^{th}\)</span> marker, we have the model below</p>
<p><span class="math display" id="eq:EG-mod">\[\begin{equation}
\mathbf{E}_k=a+\beta_l\mathbf{x}_l+e \tag{1.6}
\end{equation}\]</span></p>
<p>It consequently generates <span class="math inline">\(m\)</span> estimates of <span class="math inline">\(\hat{\beta}\)</span>, <span class="math inline">\(\hat{\sigma}_{\beta}\)</span>, and their corresponding <span class="math inline">\(p\)</span> values. The <span class="math inline">\(p\)</span> value can be converted to one-degree-of-freedom <span class="math inline">\(\chi^2_1\)</span>. Under the context of <font color="navy"><code>EigenGWAS</code></font>, the <span class="math inline">\(\chi^2_1\)</span> can be intepreted as</p>
<p><span class="math display" id="eq:EG-chi">\[\begin{equation}
4\frac{\color{red}{n_1}\color{blue}{n_2}}{n}\frac{(\color{red}{p_{1,l}}-\color{blue}{p_{2,l}})^2}{2p_lq_l}=4n\omega_1 \omega_2 F_{st}^N=nF_{st}^W \tag{1.7}
\end{equation}\]</span></p>
<p>in which <span class="math inline">\(\color{red}{n_1}\)</span> and <span class="math inline">\(\color{blue}{n_2}\)</span> are the numbers of samples at the left and right side of “0” on the eigenvector (see the figure below), and <span class="math inline">\(\omega_g=\frac{n_g}{n}\)</span> the proportion of a subgroup in the sample (<span class="math inline">\(g=2\)</span> in EigenGWAS analysis), and <span class="math inline">\(\color{red}{p_{1,l}}\)</span> and <span class="math inline">\(\color{blue}{p_{2,l}}\)</span> are the frequencies of the reference allele in two subgroups, respectively. <span class="math inline">\(F_{st}^N=\frac{(\color{red}{p_{1,l}}-\color{blue}{p_{2,l}})^2}{2p_lq_l}\)</span> and <span class="math inline">\(F_{st}^W=2\frac{\sum_{g=1}^2\omega_g(p_{g,l}-p_l)^2}{p_lq_l}\)</span>. It is obviously that when <span class="math inline">\(n_1=n_2=\frac{n}{2}\)</span> the test statistic reaches its maximum.</p>
<pre><code>## [1]  100 1000
## [1] &quot;Ne= 100 Me= 1080.91076380488 given N= 100 and M= 1000&quot;</code></pre>
<p><img src="EigenGWAS_files/figure-html/eigen-1.png" width="672" /></p>
<div id="lambda_gc-correction" class="section level3">
<h3><span class="header-section-number">1.2.1</span> <span class="math inline">\(\lambda_{GC}\)</span> correction</h3>
<p>Given the median, denoted as <span class="math inline">\(\chi^2_{1,median(p)}\)</span>, of the <span class="math inline">\(m\)</span> <span class="math inline">\(\chi^2_1\)</span> values observed, we can define <span class="math inline">\(\lambda_{GC}=\chi^2_{1,median(p)}/\chi^2_{1,0.5}\)</span>, in which <span class="math inline">\(\chi^2_{1,0.5}=0.455\)</span>. We further use subscript <span class="math inline">\(k\)</span> to denote <span class="math inline">\(\lambda_{GC_k}\)</span> the one that is estimated from the EigenGWAS analysis of <span class="math inline">\(\mathbf{E}_k\)</span>, as shown <a href="index.html#eq:EG-mod">(1.6)</a>.</p>
<p>After technical correction, correspondingly</p>
<p><span class="math display" id="eq:EG-chi-adj">\[\begin{equation}
\tilde\chi^2_1=\chi^2_1/\lambda_{GC_k} \tag{1.8}
\end{equation}\]</span></p>
<p>a correction of the test statistic. Compared with its original form, the correction has several implications</p>
<ul>
<li><p>Statistically, as <a href="index.html#eq:EG-mod">(1.6)</a> has its response variable from <span class="math inline">\(\mathbf{X}\)</span>, the correction removes its overfitting.</p></li>
<li><p>Genetically, it corrects for genetic drift such as soaked in <span class="math inline">\(\mathbf{E}_1\)</span>. Here the quantity of the genetic drift is measured by the median of the <span class="math inline">\(m\)</span> <span class="math inline">\(\chi^2_1\)</span> values observed.</p></li>
<li><p><span class="math inline">\(\lambda_{GC}\)</span> parameter has long been proposed in controlling population structure in GWAS, but finds its best fit in <font color="navy"><code>EigenGWAS</code></font>.</p></li>
</ul>
<p><img src="EigenGWAS_files/figure-html/gc-1.png" width="672" /></p>
</div>
<div id="threshold-for-eigengwas" class="section level3">
<h3><span class="header-section-number">1.2.2</span> Threshold for EigenGWAS</h3>
<p>As shown above, <font color="navy"><code>EigenGWAS</code></font> is a linear model framework nearly identical to the conventional GWAS, and Bonferroni correction, such as <span class="math inline">\(\alpha/m\)</span>, can be used to set the threshold at the significance level <span class="math inline">\(\alpha\)</span>, such as <span class="math inline">\(\alpha=0.05\)</span>. The loci that exceeds the threshold are under selection.</p>
</div>
</div>
<div id="connection-to-singular-value-decomposition" class="section level2">
<h2><span class="header-section-number">1.3</span> Connection to singular value decomposition</h2>
<p>Singular value decomposition ((see wiki for more details](<a href="https://en.wikipedia.org/wiki/Singular_value_decomposition" class="uri">https://en.wikipedia.org/wiki/Singular_value_decomposition</a>)) can decompose the matrix <span class="math inline">\(\mathbf{X}\)</span> into
<span class="math display" id="eq:SVD">\[\begin{equation}
\mathbf{X}=\mathbf{U\Sigma V} \tag{1.9}
\end{equation}\]</span>
in which</p>
<ul>
<li><p><span class="math inline">\(\mathbf{U}\)</span> is an <span class="math inline">\(n\times n\)</span> unitary matrix, corresponding to individual-level loading for each sample. <span class="math inline">\(\mathbf{U}\)</span> is also called left-singular vectors.</p></li>
<li><p><span class="math inline">\(\mathbf{\Sigma}\)</span> an <span class="math inline">\(n\times m\)</span> diagonal matrix,</p></li>
<li><p><span class="math inline">\(\mathbf{V}\)</span> an <span class="math inline">\(m\times m\)</span> unitary matrix, corresponding to SNP-level loading. <span class="math inline">\(\mathbf{U}\)</span> is also called right-singular vectors.</p></li>
</ul>
<p>Due to the transformation between <span class="math inline">\(\mathbf{U}\)</span> and <span class="math inline">\(\mathbf{V}\)</span>, it has
<span class="math display" id="eq:svd">\[\begin{equation}
\mathbf{U}=\mathbf{\tilde{X}V\Sigma}^{-1} \tag{1.10}
\end{equation}\]</span>
The right side of <a href="index.html#eq:svd">(1.10)</a> can be unfolded as <span class="math inline">\(\frac{\sqrt{m}}{\sqrt{\Sigma_k}}\mathbf{\tilde{x}}_l\mathbf{v}_k\)</span>, following <span class="math inline">\(N(0,1)\)</span>, and taking square of it leads to</p>
<p><span class="math display" id="eq:chi-svd">\[\begin{equation}
(\frac{\sqrt{m}}{\sqrt{\Sigma_k}}\mathbf{\tilde{x}}_l\mathbf{v}_k)^2 \sim \tilde{\chi}^2_{1,svd} \tag{1.11}
\end{equation}\]</span></p>
<p>It brings out an interesting comparison between <a href="index.html#eq:EG-chi-adj">(1.8)</a> and <a href="index.html#eq:svd">(1.10)</a>,
and exists subtle difference, for instance <span class="math inline">\(E_1\)</span>, in their correction using <span class="math inline">\(\lambda_{GC}\)</span> and <span class="math inline">\(\Sigma_1\)</span>. When the population divergency is due to genetic drift and a small proportion of loci are under selection, <span class="math inline">\(\Sigma_1 \gt \lambda_{GC}\)</span>. It is because <span class="math inline">\(\Sigma_1\)</span> is the mean of the <span class="math inline">\(\chi^2_1\)</span> from <a href="index.html#eq:EG-chi">(1.7)</a>, but <span class="math inline">\(\lambda_{GC}\)</span> is the median of it. Given a population under selection, such as selection sweep, it does have <span class="math inline">\(\Sigma_1 \gt \lambda_{GC}\)</span>, as demonstrated in <a href="http://rpubs.com/gc5k/EigenGWAScore">simulation</a>.</p>
<p>So, using linear model system such as <font color="navy"><code>EigenGWAS</code></font> brings more flexibility, as well as improved statistical power comparing with direct <code>left-to-right singular vector</code> transformation in SVD <a href="index.html#eq:svd">(1.10)</a>.</p>
<p>However, in this transformation, eigenvalue is involved, as would be show. It will reduce the statistical power for <font color="navy"><code>EigenGWAS</code></font>. In theory and practice, with type-I error rate well-controlled, <span class="math inline">\(\tilde{\chi}^2_1\)</span> <a href="index.html#eq:EG-chi-adj">(1.8)</a> is greater than <span class="math inline">\(\tilde{\chi}^2_{1.svd}\)</span> <a href="index.html#eq:svd">(1.10)</a>.</p>
</div>
<div id="intepretation" class="section level2">
<h2><span class="header-section-number">1.4</span> Intepretation</h2>
<div id="f_st" class="section level3">
<h3><span class="header-section-number">1.4.1</span> <span class="math inline">\(F_{st}\)</span></h3>
<p><code>EigenGWAS</code> provides an unsupervised realization for <span class="math inline">\(F_{st}\)</span> the conventional implementation of that assumes the underlying subgroups are known. In practice, however, it is often unclear how the possible subgroups should be defined. In contrast, <code>EigenGWAS</code> implicitly partitions the samples into two subgroups at each eigenvector at zero point on the coordinate. No guarantee that this strategy optimal, it seems a sound attempt when the group information is missing.</p>
</div>
<div id="classic-mechanic-intepretation" class="section level3">
<h3><span class="header-section-number">1.4.2</span> Classic mechanic intepretation</h3>
<p>The early framework of population genetics indebts a lot for theoretical physics (<a href="https://en.wikipedia.org/wiki/Ronald_Fisher">Ronald Fisher</a> studied astronomy). As a linear model system, <font color="navy"><code>EigenGWAS</code></font> also finds its analogue in Newtown’s first and the second <a href="https://en.wikipedia.org/wiki/Newton%27s_laws_of_motion">laws for classical mechanics</a>. The first law states</p>
<blockquote>
<p>“In an inertial frame of reference, an object either remains at rest or continues to move at a constant velocity, unless acted upon by a force.”</p>
</blockquote>
<p>In population genetics, it can be seemed as genetic drift that is constantly driving the a pair of population apart from each other, and its velocity can be quantified by a binomial distribution as <span class="math inline">\(\frac{pq}{\tilde{n}_e}\)</span>.</p>
<p>The second law stats</p>
<blockquote>
<p>“In an inertial frame of reference, the vector sum of the forces <span class="math inline">\(F\)</span> on an object is equal to the mass m of that object multiplied by the acceleration a of the object: <span class="math inline">\(F = ma\)</span>.”</p>
</blockquote>
<p>The selection can drive a genomic region run against its reference population at a velocity greater than <span class="math inline">\(\frac{pq}{\tilde{n}_e}\)</span>.</p>
<p>Although, this interpretation works even for domesticated populations, its proportion of the markers divergent by genetic drift and selection would be very different.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="protocols.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["EigenGWAS.pdf", "EigenGWAS.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
