<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>EigenGWAS theory and application</title>
  <meta name="description" content="EigenGWAS Project.">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="EigenGWAS theory and application" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="EigenGWAS Project." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="EigenGWAS theory and application" />
  
  <meta name="twitter:description" content="EigenGWAS Project." />
  

<meta name="author" content="Guo-Bo Chen [chen.guobo@foxmail.com]">


<meta name="date" content="2019-05-31">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="references.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EigenGWAS core</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> EigenGWAS basis</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#genetic-relatedness-matrix-mathbfg"><i class="fa fa-check"></i><b>1.1</b> Genetic relatedness matrix <span class="math inline">\(\mathbf{G}\)</span></a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#statistical-properties-of-mathbfg"><i class="fa fa-check"></i><b>1.1.1</b> Statistical properties of <span class="math inline">\(\mathbf{G}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#eigengwas-linear-model"><i class="fa fa-check"></i><b>1.2</b> EigenGWAS linear model</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#lambda_gc-correction"><i class="fa fa-check"></i><b>1.2.1</b> <span class="math inline">\(\lambda_{GC}\)</span> correction</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#threshold-for-eigengwas"><i class="fa fa-check"></i><b>1.2.2</b> Threshold for EigenGWAS</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#connection-to-singular-value-decomposition"><i class="fa fa-check"></i><b>1.3</b> Connection to singular value decomposition</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#intepretation"><i class="fa fa-check"></i><b>1.4</b> Intepretation</a><ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#f_st"><i class="fa fa-check"></i><b>1.4.1</b> <span class="math inline">\(F_{st}\)</span></a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#classic-mechanic-intepretation"><i class="fa fa-check"></i><b>1.4.2</b> Classic mechanic intepretation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="protocols.html"><a href="protocols.html"><i class="fa fa-check"></i><b>2</b> Protocols</a><ul>
<li class="chapter" data-level="2.1" data-path="protocols.html"><a href="protocols.html#protocols-for-selection"><i class="fa fa-check"></i><b>2.1</b> Protocols for selection</a></li>
<li class="chapter" data-level="2.2" data-path="protocols.html"><a href="protocols.html#protocal-for-eigengwas"><i class="fa fa-check"></i><b>2.2</b> Protocal for EigenGWAS</a></li>
<li class="chapter" data-level="2.3" data-path="protocols.html"><a href="protocols.html#protocol-for-predicted-eigenvectors"><i class="fa fa-check"></i><b>2.3</b> Protocol for predicted eigenvectors</a></li>
<li class="chapter" data-level="2.4" data-path="protocols.html"><a href="protocols.html#nucleotide-diversity"><i class="fa fa-check"></i><b>2.4</b> Nucleotide diversity</a></li>
<li class="chapter" data-level="2.5" data-path="protocols.html"><a href="protocols.html#other-test-statistics-for-selection"><i class="fa fa-check"></i><b>2.5</b> Other test statistics for selection</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="resequencing-studies.html"><a href="resequencing-studies.html"><i class="fa fa-check"></i><b>3</b> Resequencing studies</a><ul>
<li class="chapter" data-level="3.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#technical-review"><i class="fa fa-check"></i><b>3.1</b> Technical review</a><ul>
<li class="chapter" data-level="3.1.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#simulation"><i class="fa fa-check"></i><b>3.1.1</b> Simulation</a></li>
<li class="chapter" data-level="3.1.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#ngs"><i class="fa fa-check"></i><b>3.1.2</b> NGS</a></li>
<li class="chapter" data-level="3.1.3" data-path="resequencing-studies.html"><a href="resequencing-studies.html#chip-data"><i class="fa fa-check"></i><b>3.1.3</b> Chip data</a></li>
<li class="chapter" data-level="3.1.4" data-path="resequencing-studies.html"><a href="resequencing-studies.html#gbs"><i class="fa fa-check"></i><b>3.1.4</b> GBS</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#drop-hwe-test"><i class="fa fa-check"></i><b>3.2</b> Drop HWE test</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html"><i class="fa fa-check"></i><b>4</b> Simulating population structure</a><ul>
<li class="chapter" data-level="4.1" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#genetic-drift"><i class="fa fa-check"></i><b>4.1</b> Genetic drift</a></li>
<li class="chapter" data-level="4.2" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#discrete-populations"><i class="fa fa-check"></i><b>4.2</b> Discrete populations</a></li>
<li class="chapter" data-level="4.3" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#admixture-populations"><i class="fa fa-check"></i><b>4.3</b> Admixture populations</a></li>
<li class="chapter" data-level="4.4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#homo-heteogeneous-f_st"><i class="fa fa-check"></i><b>4.4</b> Homo &amp; Heteogeneous <span class="math inline">\(F_{st}\)</span></a></li>
<li class="chapter" data-level="4.5" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#wishart-distribution"><i class="fa fa-check"></i><b>4.5</b> Wishart distribution</a></li>
<li class="chapter" data-level="4.6" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#tracy-widom-distribution"><i class="fa fa-check"></i><b>4.6</b> Tracy-Widom distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>5</b> Data analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="data-analysis.html"><a href="data-analysis.html#on-site-examples"><i class="fa fa-check"></i><b>5.1</b> On-site examples</a><ul>
<li class="chapter" data-level="5.1.1" data-path="data-analysis.html"><a href="data-analysis.html#arabdiopsis"><i class="fa fa-check"></i><b>5.1.1</b> Arabdiopsis</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-analysis.html"><a href="data-analysis.html#k-rice"><i class="fa fa-check"></i><b>5.1.2</b> 3K rice</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-analysis.html"><a href="data-analysis.html#als"><i class="fa fa-check"></i><b>5.1.3</b> ALS</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-analysis.html"><a href="data-analysis.html#gf"><i class="fa fa-check"></i><b>5.1.4</b> GF</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-analysis.html"><a href="data-analysis.html#uk-birds"><i class="fa fa-check"></i><b>5.1.5</b> UK birds</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-analysis.html"><a href="data-analysis.html#darwins-finches"><i class="fa fa-check"></i><b>5.1.6</b> Darwin’s finches</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-analysis.html"><a href="data-analysis.html#gbs-maize"><i class="fa fa-check"></i><b>5.1.7</b> GBS Maize</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-analysis.html"><a href="data-analysis.html#uk-biobank"><i class="fa fa-check"></i><b>5.1.8</b> UK Biobank</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-analysis.html"><a href="data-analysis.html#public-datahub-neo"><i class="fa fa-check"></i><b>5.2</b> Public datahub (NEO)</a></li>
<li class="chapter" data-level="5.3" data-path="data-analysis.html"><a href="data-analysis.html#meta-scale"><i class="fa fa-check"></i><b>5.3</b> Meta-scale</a><ul>
<li class="chapter" data-level="5.3.1" data-path="data-analysis.html"><a href="data-analysis.html#n_e"><i class="fa fa-check"></i><b>5.3.1</b> <span class="math inline">\(n_e\)</span></a></li>
<li class="chapter" data-level="5.3.2" data-path="data-analysis.html"><a href="data-analysis.html#m_e"><i class="fa fa-check"></i><b>5.3.2</b> <span class="math inline">\(m_e\)</span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a><ul>
<li class="chapter" data-level="6.1" data-path="conclusion.html"><a href="conclusion.html#statistical-power"><i class="fa fa-check"></i><b>6.1</b> Statistical power</a></li>
<li class="chapter" data-level="6.2" data-path="conclusion.html"><a href="conclusion.html#selection-pattern"><i class="fa fa-check"></i><b>6.2</b> Selection pattern</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="app-1.html"><a href="app-1.html"><i class="fa fa-check"></i><b>7</b> Appendix 1</a><ul>
<li class="chapter" data-level="7.1" data-path="app-1.html"><a href="app-1.html#app:ld"><i class="fa fa-check"></i><b>7.1</b> Notes on linkage disequilibrium</a></li>
<li class="chapter" data-level="7.2" data-path="app-1.html"><a href="app-1.html#the-maximal-value-rho_ab"><i class="fa fa-check"></i><b>7.2</b> The maximal value <span class="math inline">\(\rho_{AB}\)</span></a></li>
<li class="chapter" data-level="7.3" data-path="app-1.html"><a href="app-1.html#ld-and-recombination-fraction"><i class="fa fa-check"></i><b>7.3</b> LD and recombination fraction</a></li>
<li class="chapter" data-level="7.4" data-path="app-1.html"><a href="app-1.html#f_st-1"><i class="fa fa-check"></i><b>7.4</b> <span class="math inline">\(F_{st}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="8" data-path="notes-1.html"><a href="notes-1.html"><i class="fa fa-check"></i><b>8</b> Notes</a><ul>
<li class="chapter" data-level="8.1" data-path="notes-1.html"><a href="notes-1.html#apy"><i class="fa fa-check"></i><b>8.1</b> APY</a><ul>
<li class="chapter" data-level="8.1.1" data-path="notes-1.html"><a href="notes-1.html#a-numerical-example"><i class="fa fa-check"></i><b>8.1.1</b> A numerical example</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="notes-1.html"><a href="notes-1.html#ld-score-regression"><i class="fa fa-check"></i><b>8.2</b> LD score regression</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://gc5k.github.io" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EigenGWAS theory and application</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="notes-1" class="section level1">
<h1><span class="header-section-number">Chapter 8</span> Notes</h1>
<div id="apy" class="section level2">
<h2><span class="header-section-number">8.1</span> APY</h2>
<p>Citation: Misztal, I, 2016, Inexpensive computation of the inverse of the genomic relationship matrix in populations with small effective population size, <span class="math inline">\(Genetics\)</span>, 202:401-409</p>
<p><span class="math inline">\(G^{-1}= \left[ \begin{matrix} I&amp; -P_{cn}\\ 0&amp;I \end{matrix} \right] \left[ \begin{matrix} G_{cc}^{-1}&amp; 0\\ 0&amp;M_{nn}^{-1} \end{matrix} \right] \left[ \begin{matrix} I&amp; 0\\ -P_{nc}&amp;I \end{matrix} \right] = \left[ \begin{matrix}G_{cc}^{-1}&amp;-P_{cn}M_{nn}^{-1}\\ 0&amp;M^{-1}_{nn} \end{matrix} \right] \left[ \begin{matrix}I&amp;0\\ -P_{nc}&amp;I \end{matrix} \right]\)</span></p>
<p>in which <span class="math inline">\(G\)</span> is the numerical relationship matrix, and <span class="math inline">\(P_{cn}=G_{cc}^{-1}G_{cn}\)</span> and <span class="math inline">\(P_{nc}=G_{cn}G_{cc}^{-1}\)</span>. <span class="math inline">\(G_{cc}\)</span>, <span class="math inline">\(G_{cn}\)</span>, and <span class="math inline">\(G_{nc}\)</span> are, “core” to “core”, “core” to “non-core”, and “non-core” to “core” relationship matrix.</p>
<p><span class="math inline">\(M_{nn}=diag(G_{nn})-diag(P_{cn}^TG_{cn})\)</span></p>
<div id="a-numerical-example" class="section level3">
<h3><span class="header-section-number">8.1.1</span> A numerical example</h3>
<p><span class="math inline">\(G=\left[ \begin{matrix} 0.81 &amp; 0 &amp; 0 &amp; 0.80 &amp; -0.80\\  &amp;0.81 &amp; 0 &amp; 0.80 &amp;-0.80\\  &amp; &amp; 0.01&amp; 0 &amp; 0\\  &amp; &amp; &amp; 1.61 &amp; -1.60\\  symm. &amp; &amp; &amp; &amp; 1.61 \end{matrix} \right]\)</span></p>
<p>and <span class="math inline">\(G_{cc}^{-1}=\left[ \begin{matrix} 1.235 &amp; 0\\ 0 &amp; 1.235 \end{matrix} \right]\)</span></p>
<p><span class="math inline">\(G_{cn}=G_{nc}^T=\left [ \begin{matrix} 0 &amp; 0.80 &amp; -0.80\\ 0 &amp; 0.80 &amp; -0.80 \end{matrix} \right]\)</span></p>
<p>and</p>
<p><span class="math inline">\(P_{cn}=G_{cc}^{-1}G_{cn}=\left [ \begin{matrix} 0.00 &amp; 0.988 &amp; -0.988\\ 0.00 &amp; 0.988 &amp; -0.988 \end{matrix} \right ]\)</span></p>
<p><span class="math inline">\(M_{nn}=diag(G_{nn})-diag(P_{cn}^TG_{cn}) = \left [ \begin{matrix}0.01 &amp; &amp;\\ &amp; 1.61 &amp; \\ &amp; &amp; 1.61 \end{matrix} \right] - \left [ \begin{matrix}0.00 &amp; &amp;\\ &amp; 1.58 &amp; \\ &amp; &amp; 1.58 \end{matrix} \right]=\left [ \begin{matrix}0.01 &amp; &amp;\\ &amp; 0.03 &amp; \\ &amp; &amp; 0.03 \end{matrix} \right]\)</span></p>
<p>and</p>
<p><span class="math inline">\(G^{-1}=\left[ \begin{matrix} 66.8 &amp; 65.5 &amp; 0 &amp; -33.1 &amp; 33.1\\  &amp;66.804 &amp; 0 &amp; -33.1 &amp; 33.1\\  &amp; &amp; 100&amp; 0 &amp; 0\\  &amp; &amp; &amp; 33.6 &amp; 0\\  symm. &amp; &amp; &amp; &amp; 33.6 \end{matrix} \right]\)</span></p>
<p>This is APY inversed <span class="math inline">\(G\)</span>, and for a comparison, a regular inverse of <span class="math inline">\(G\)</span> is quit different
<span class="math inline">\(G^{-1}_{reg}=\left[ \begin{matrix} 40.6 &amp; 39.4 &amp; 0 &amp; -19.9 &amp; 19.9\\  &amp;40.6 &amp; 0 &amp; -19.9 &amp; 19.9\\  &amp; &amp; 100&amp; 0 &amp; 0\\  &amp; &amp; &amp; 60.0 &amp; 39.9\\  symm. &amp; &amp; &amp; &amp; 60.0 \end{matrix} \right]\)</span></p>
</div>
</div>
<div id="ld-score-regression" class="section level2">
<h2><span class="header-section-number">8.2</span> LD score regression</h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="references.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["EigenGWAS.pdf", "EigenGWAS.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
