<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Notes | EigenGWAS theory and application</title>
  <meta name="description" content="EigenGWAS Project." />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Notes | EigenGWAS theory and application" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="EigenGWAS Project." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Notes | EigenGWAS theory and application" />
  
  <meta name="twitter:description" content="EigenGWAS Project." />
  

<meta name="author" content="Guo-Bo Chen [chen.guobo@foxmail.com]" />


<meta name="date" content="2020-02-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="references.html"/>
<link rel="next" href="IBD-fam.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EigenGWAS core</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> EigenGWAS basis</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#genetic-relatedness-matrix-mathbfg"><i class="fa fa-check"></i><b>1.1</b> Genetic relatedness matrix <span class="math inline">\(\mathbf{G}\)</span></a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#statistical-properties-of-mathbfg"><i class="fa fa-check"></i><b>1.1.1</b> Statistical properties of <span class="math inline">\(\mathbf{G}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#eigengwas-linear-model"><i class="fa fa-check"></i><b>1.2</b> EigenGWAS linear model</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#lambda_gc-correction"><i class="fa fa-check"></i><b>1.2.1</b> <span class="math inline">\(\lambda_{GC}\)</span> correction</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#threshold-for-eigengwas"><i class="fa fa-check"></i><b>1.2.2</b> Threshold for EigenGWAS</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#connection-to-singular-value-decomposition"><i class="fa fa-check"></i><b>1.3</b> Connection to singular value decomposition</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#intepretation"><i class="fa fa-check"></i><b>1.4</b> Intepretation</a><ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#f_st"><i class="fa fa-check"></i><b>1.4.1</b> <span class="math inline">\(F_{st}\)</span></a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#classic-mechanic-intepretation"><i class="fa fa-check"></i><b>1.4.2</b> Classic mechanic intepretation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="protocols.html"><a href="protocols.html"><i class="fa fa-check"></i><b>2</b> Protocols</a><ul>
<li class="chapter" data-level="2.1" data-path="protocols.html"><a href="protocols.html#protocols-for-selection"><i class="fa fa-check"></i><b>2.1</b> Protocols for selection</a></li>
<li class="chapter" data-level="2.2" data-path="protocols.html"><a href="protocols.html#protocal-for-eigengwas"><i class="fa fa-check"></i><b>2.2</b> Protocal for EigenGWAS</a></li>
<li class="chapter" data-level="2.3" data-path="protocols.html"><a href="protocols.html#protocol-for-predicted-eigenvectors"><i class="fa fa-check"></i><b>2.3</b> Protocol for predicted eigenvectors</a></li>
<li class="chapter" data-level="2.4" data-path="protocols.html"><a href="protocols.html#nucleotide-diversity"><i class="fa fa-check"></i><b>2.4</b> Nucleotide diversity</a></li>
<li class="chapter" data-level="2.5" data-path="protocols.html"><a href="protocols.html#other-test-statistics-for-selection"><i class="fa fa-check"></i><b>2.5</b> Other test statistics for selection</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="resequencing-studies.html"><a href="resequencing-studies.html"><i class="fa fa-check"></i><b>3</b> Resequencing studies</a><ul>
<li class="chapter" data-level="3.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#technical-review"><i class="fa fa-check"></i><b>3.1</b> Technical review</a><ul>
<li class="chapter" data-level="3.1.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#simulation"><i class="fa fa-check"></i><b>3.1.1</b> Simulation</a></li>
<li class="chapter" data-level="3.1.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#ngs"><i class="fa fa-check"></i><b>3.1.2</b> NGS</a></li>
<li class="chapter" data-level="3.1.3" data-path="resequencing-studies.html"><a href="resequencing-studies.html#chip-data"><i class="fa fa-check"></i><b>3.1.3</b> Chip data</a></li>
<li class="chapter" data-level="3.1.4" data-path="resequencing-studies.html"><a href="resequencing-studies.html#gbs"><i class="fa fa-check"></i><b>3.1.4</b> GBS</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#drop-hwe-test"><i class="fa fa-check"></i><b>3.2</b> Drop HWE test</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html"><i class="fa fa-check"></i><b>4</b> Simulating population structure</a><ul>
<li class="chapter" data-level="4.1" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#genetic-drift"><i class="fa fa-check"></i><b>4.1</b> Genetic drift</a></li>
<li class="chapter" data-level="4.2" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#discrete-populations"><i class="fa fa-check"></i><b>4.2</b> Discrete populations</a></li>
<li class="chapter" data-level="4.3" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#admixture-populations"><i class="fa fa-check"></i><b>4.3</b> Admixture populations</a></li>
<li class="chapter" data-level="4.4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#homo-heteogeneous-f_st"><i class="fa fa-check"></i><b>4.4</b> Homo &amp; Heteogeneous <span class="math inline">\(F_{st}\)</span></a></li>
<li class="chapter" data-level="4.5" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#wishart-distribution"><i class="fa fa-check"></i><b>4.5</b> Wishart distribution</a></li>
<li class="chapter" data-level="4.6" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#tracy-widom-distribution"><i class="fa fa-check"></i><b>4.6</b> Tracy-Widom distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>5</b> Data analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="data-analysis.html"><a href="data-analysis.html#on-site-examples"><i class="fa fa-check"></i><b>5.1</b> On-site examples</a><ul>
<li class="chapter" data-level="5.1.1" data-path="data-analysis.html"><a href="data-analysis.html#arabdiopsis"><i class="fa fa-check"></i><b>5.1.1</b> Arabdiopsis</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-analysis.html"><a href="data-analysis.html#k-rice"><i class="fa fa-check"></i><b>5.1.2</b> 3K rice</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-analysis.html"><a href="data-analysis.html#als"><i class="fa fa-check"></i><b>5.1.3</b> ALS</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-analysis.html"><a href="data-analysis.html#gf"><i class="fa fa-check"></i><b>5.1.4</b> GF</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-analysis.html"><a href="data-analysis.html#uk-birds"><i class="fa fa-check"></i><b>5.1.5</b> UK birds</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-analysis.html"><a href="data-analysis.html#darwins-finches"><i class="fa fa-check"></i><b>5.1.6</b> Darwin’s finches</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-analysis.html"><a href="data-analysis.html#gbs-maize"><i class="fa fa-check"></i><b>5.1.7</b> GBS Maize</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-analysis.html"><a href="data-analysis.html#uk-biobank"><i class="fa fa-check"></i><b>5.1.8</b> UK Biobank</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-analysis.html"><a href="data-analysis.html#public-datahub-neo"><i class="fa fa-check"></i><b>5.2</b> Public datahub (NEO)</a></li>
<li class="chapter" data-level="5.3" data-path="data-analysis.html"><a href="data-analysis.html#meta-scale"><i class="fa fa-check"></i><b>5.3</b> Meta-scale</a><ul>
<li class="chapter" data-level="5.3.1" data-path="data-analysis.html"><a href="data-analysis.html#n_e"><i class="fa fa-check"></i><b>5.3.1</b> <span class="math inline">\(n_e\)</span></a></li>
<li class="chapter" data-level="5.3.2" data-path="data-analysis.html"><a href="data-analysis.html#m_e"><i class="fa fa-check"></i><b>5.3.2</b> <span class="math inline">\(m_e\)</span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a><ul>
<li class="chapter" data-level="6.1" data-path="conclusion.html"><a href="conclusion.html#statistical-power"><i class="fa fa-check"></i><b>6.1</b> Statistical power</a></li>
<li class="chapter" data-level="6.2" data-path="conclusion.html"><a href="conclusion.html#selection-pattern"><i class="fa fa-check"></i><b>6.2</b> Selection pattern</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="app-1.html"><a href="app-1.html"><i class="fa fa-check"></i><b>7</b> Appendix 1</a><ul>
<li class="chapter" data-level="7.1" data-path="app-1.html"><a href="app-1.html#app:ld"><i class="fa fa-check"></i><b>7.1</b> Notes on linkage disequilibrium</a></li>
<li class="chapter" data-level="7.2" data-path="app-1.html"><a href="app-1.html#the-maximal-value-rho_ab"><i class="fa fa-check"></i><b>7.2</b> The maximal value <span class="math inline">\(\rho_{AB}\)</span></a></li>
<li class="chapter" data-level="7.3" data-path="app-1.html"><a href="app-1.html#ld-and-recombination-fraction"><i class="fa fa-check"></i><b>7.3</b> LD and recombination fraction</a></li>
<li class="chapter" data-level="7.4" data-path="app-1.html"><a href="app-1.html#f_st-1"><i class="fa fa-check"></i><b>7.4</b> <span class="math inline">\(F_{st}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="8" data-path="notes-1.html"><a href="notes-1.html"><i class="fa fa-check"></i><b>8</b> Notes</a><ul>
<li class="chapter" data-level="8.1" data-path="notes-1.html"><a href="notes-1.html#apy"><i class="fa fa-check"></i><b>8.1</b> APY</a><ul>
<li class="chapter" data-level="8.1.1" data-path="notes-1.html"><a href="notes-1.html#a-numerical-example"><i class="fa fa-check"></i><b>8.1.1</b> A numerical example</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="notes-1.html"><a href="notes-1.html#ld-score-regression"><i class="fa fa-check"></i><b>8.2</b> LD score regression</a></li>
<li class="chapter" data-level="8.3" data-path="notes-1.html"><a href="notes-1.html#matrix-inversion-wiki"><i class="fa fa-check"></i><b>8.3</b> Matrix inversion <span>wiki</span></a><ul>
<li class="chapter" data-level="8.3.1" data-path="notes-1.html"><a href="notes-1.html#x-2"><i class="fa fa-check"></i><b>8.3.1</b> 2 X 2</a></li>
<li class="chapter" data-level="8.3.2" data-path="notes-1.html"><a href="notes-1.html#x-3"><i class="fa fa-check"></i><b>8.3.2</b> 3 X 3</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="IBD-fam.html"><a href="IBD-fam.html"><i class="fa fa-check"></i><b>9</b> IBD notes</a><ul>
<li class="chapter" data-level="9.1" data-path="IBD-fam.html"><a href="IBD-fam.html#he-association"><i class="fa fa-check"></i><b>9.1</b> HE association</a></li>
<li class="chapter" data-level="9.2" data-path="IBD-fam.html"><a href="IBD-fam.html#ibd-table"><i class="fa fa-check"></i><b>9.2</b> IBD table</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="app-2.html"><a href="app-2.html"><i class="fa fa-check"></i><b>10</b> Appendix quick pca</a><ul>
<li class="chapter" data-level="10.1" data-path="app-2.html"><a href="app-2.html#quck-pca-method-for-snp-matrix"><i class="fa fa-check"></i><b>10.1</b> Quck PCA method for snp matrix</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://gc5k.github.io" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EigenGWAS theory and application</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="notes-1" class="section level1">
<h1><span class="header-section-number">Chapter 8</span> Notes</h1>
<div id="apy" class="section level2">
<h2><span class="header-section-number">8.1</span> APY</h2>
<p>Citation: Misztal, I, 2016, Inexpensive computation of the inverse of the genomic relationship matrix in populations with small effective population size, <span class="math inline">\(Genetics\)</span>, 202:401-409.</p>
<p>A much original idea of APY resembles Quaas and Pollak (J Anim Sci, 1980, 51:1277-87), an example of which can be found in Lynch&amp;Walsh page 759-61.</p>
<p><span class="math inline">\(G^{-1}= \left[ \begin{matrix} I&amp; -P_{cn}\\ 0&amp;I \end{matrix} \right] \left[ \begin{matrix} G_{cc}^{-1}&amp; 0\\ 0&amp;M_{nn}^{-1} \end{matrix} \right] \left[ \begin{matrix} I&amp; 0\\ -P_{nc}&amp;I \end{matrix} \right] = \left[ \begin{matrix}G_{cc}^{-1}&amp;-P_{cn}M_{nn}^{-1}\\ 0&amp;M^{-1}_{nn} \end{matrix} \right] \left[ \begin{matrix}I&amp;0\\ -P_{nc}&amp;I \end{matrix} \right]\)</span></p>
<p>in which <span class="math inline">\(G\)</span> is the numerical relationship matrix, and <span class="math inline">\(P_{cn}=G_{cc}^{-1}G_{cn}\)</span> and <span class="math inline">\(P_{nc}=G_{nc}G_{cc}^{-1}\)</span>. <span class="math inline">\(G_{cc}\)</span>, <span class="math inline">\(G_{cn}\)</span>, and <span class="math inline">\(G_{nc}\)</span> (<span class="math inline">\(G_{cn}=G_{nc}\)</span>) are, “core” to “core”, “core” to “non-core”, and “non-core” to “core” relationship matrix.</p>
<p><span class="math inline">\(M_{nn}=diag(G_{nn})-diag(P_{cn}^TG_{cn})\)</span></p>
<div id="a-numerical-example" class="section level3">
<h3><span class="header-section-number">8.1.1</span> A numerical example</h3>
<p>This example is taken from appendix in Misztal’s paper.</p>
<p><span class="math inline">\(G=\left[ \begin{matrix} 0.81 &amp; 0 &amp; 0 &amp; 0.80 &amp; -0.80\\  &amp;0.81 &amp; 0 &amp; 0.80 &amp;-0.80\\  &amp; &amp; 0.01&amp; 0 &amp; 0\\  &amp; &amp; &amp; 1.61 &amp; -1.60\\  symm. &amp; &amp; &amp; &amp; 1.61 \end{matrix} \right]\)</span></p>
<p>and <span class="math inline">\(G_{cc}^{-1}=\left[ \begin{matrix} 1.235 &amp; 0\\ 0 &amp; 1.235 \end{matrix} \right]\)</span></p>
<p><span class="math inline">\(G_{cn}=G_{nc}^T=\left [ \begin{matrix} 0 &amp; 0.80 &amp; -0.80\\ 0 &amp; 0.80 &amp; -0.80 \end{matrix} \right]\)</span></p>
<p>and</p>
<p><span class="math inline">\(P_{cn}=G_{cc}^{-1}G_{cn}=\left [ \begin{matrix} 0.00 &amp; 0.988 &amp; -0.988\\ 0.00 &amp; 0.988 &amp; -0.988 \end{matrix} \right ]\)</span></p>
<p><span class="math inline">\(M_{nn}=diag(G_{nn})-diag(P_{cn}^TG_{cn}) = \left [ \begin{matrix}0.01 &amp; &amp;\\ &amp; 1.61 &amp; \\ &amp; &amp; 1.61 \end{matrix} \right] - \left [ \begin{matrix}0.00 &amp; &amp;\\ &amp; 1.58 &amp; \\ &amp; &amp; 1.58 \end{matrix} \right]=\left [ \begin{matrix}0.01 &amp; &amp;\\ &amp; 0.03 &amp; \\ &amp; &amp; 0.03 \end{matrix} \right]\)</span></p>
<p>and</p>
<p><span class="math inline">\(G^{-1}=\left[ \begin{matrix} 66.8 &amp; 65.5 &amp; 0 &amp; -33.1 &amp; 33.1\\  &amp;66.804 &amp; 0 &amp; -33.1 &amp; 33.1\\  &amp; &amp; 100&amp; 0 &amp; 0\\  &amp; &amp; &amp; 33.6 &amp; 0\\  symm. &amp; &amp; &amp; &amp; 33.6 \end{matrix} \right]\)</span></p>
<p>This is APY inversed <span class="math inline">\(G\)</span>, and for a comparison, a regular inverse of <span class="math inline">\(G\)</span> is quit different
<span class="math inline">\(G^{-1}_{reg}=\left[ \begin{matrix} 40.6 &amp; 39.4 &amp; 0 &amp; -19.9 &amp; 19.9\\  &amp;40.6 &amp; 0 &amp; -19.9 &amp; 19.9\\  &amp; &amp; 100&amp; 0 &amp; 0\\  &amp; &amp; &amp; 60.0 &amp; 39.9\\  symm. &amp; &amp; &amp; &amp; 60.0 \end{matrix} \right]\)</span></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1">GRM_C=<span class="kw">matrix</span>(<span class="kw">c</span>(<span class="fl">0.81</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">0.81</span>), <span class="dv">2</span>, <span class="dv">2</span>, <span class="dt">byrow =</span> T)</a>
<a class="sourceLine" id="cb9-2" title="2">GRM_N=<span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dv">3</span>, <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb9-3" title="3"><span class="kw">diag</span>(GRM_N)=<span class="kw">c</span>(.<span class="dv">01</span>, <span class="fl">1.61</span>, <span class="fl">1.61</span>)</a>
<a class="sourceLine" id="cb9-4" title="4">GRM_N[<span class="dv">2</span>,<span class="dv">3</span>]=GRM_N[<span class="dv">3</span>,<span class="dv">2</span>]=<span class="op">-</span><span class="fl">1.6</span></a>
<a class="sourceLine" id="cb9-5" title="5">GRM_CN=<span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb9-6" title="6">GRM_CN[,<span class="dv">2</span>]=<span class="fl">0.8</span></a>
<a class="sourceLine" id="cb9-7" title="7">GRM_CN[,<span class="dv">3</span>]=<span class="op">-</span><span class="fl">0.8</span></a>
<a class="sourceLine" id="cb9-8" title="8"></a>
<a class="sourceLine" id="cb9-9" title="9">GRM=<span class="kw">rbind</span>(<span class="kw">cbind</span>(GRM_C, GRM_CN), <span class="kw">cbind</span>(<span class="kw">t</span>(GRM_CN), GRM_N))</a>
<a class="sourceLine" id="cb9-10" title="10"><span class="kw">print</span>(GRM)</a></code></pre></div>
<pre><code>##       [,1]  [,2] [,3]  [,4]  [,5]
## [1,]  0.81  0.00 0.00  0.80 -0.80
## [2,]  0.00  0.81 0.00  0.80 -0.80
## [3,]  0.00  0.00 0.01  0.00  0.00
## [4,]  0.80  0.80 0.00  1.61 -1.60
## [5,] -0.80 -0.80 0.00 -1.60  1.61</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1"><span class="kw">print</span>(<span class="kw">solve</span>(GRM)) <span class="co">#original</span></a></code></pre></div>
<pre><code>##           [,1]      [,2] [,3]      [,4]     [,5]
## [1,]  40.64222  39.40765    0 -19.95012 19.95012
## [2,]  39.40765  40.64222    0 -19.95012 19.95012
## [3,]   0.00000   0.00000  100   0.00000  0.00000
## [4,] -19.95012 -19.95012    0  60.09975 39.90025
## [5,]  19.95012  19.95012    0  39.90025 60.09975</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1">IC=<span class="kw">solve</span>(GRM_C)</a>
<a class="sourceLine" id="cb13-2" title="2">Pcn=IC <span class="op">%*%</span><span class="st"> </span>GRM_CN</a>
<a class="sourceLine" id="cb13-3" title="3">Pnc=<span class="kw">t</span>(GRM_CN) <span class="op">%*%</span><span class="st"> </span>IC</a>
<a class="sourceLine" id="cb13-4" title="4">Pcn_Gcn=<span class="kw">t</span>(Pcn) <span class="op">%*%</span><span class="st"> </span>GRM_CN</a>
<a class="sourceLine" id="cb13-5" title="5">Mnn=<span class="kw">diag</span>(<span class="kw">diag</span>(GRM_N)<span class="op">-</span><span class="kw">diag</span>(Pcn_Gcn), <span class="dt">nrow=</span><span class="kw">nrow</span>(GRM_N), <span class="dt">ncol=</span><span class="kw">nrow</span>(GRM_N))</a>
<a class="sourceLine" id="cb13-6" title="6">IMnn=<span class="kw">solve</span>(Mnn)</a>
<a class="sourceLine" id="cb13-7" title="7"></a>
<a class="sourceLine" id="cb13-8" title="8">v1_<span class="dv">1</span>=<span class="kw">rbind</span>(IC, <span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow=</span><span class="kw">nrow</span>(GRM_N), <span class="dt">ncol=</span><span class="kw">ncol</span>(GRM_C)))</a>
<a class="sourceLine" id="cb13-9" title="9">v1_<span class="dv">2</span>=<span class="kw">rbind</span>(<span class="op">-</span><span class="dv">1</span><span class="op">*</span>Pcn<span class="op">%*%</span>IMnn, IMnn)</a>
<a class="sourceLine" id="cb13-10" title="10">V1=<span class="kw">cbind</span>(v1_<span class="dv">1</span>, v1_<span class="dv">2</span>)</a>
<a class="sourceLine" id="cb13-11" title="11"></a>
<a class="sourceLine" id="cb13-12" title="12">v2_<span class="dv">1</span>=<span class="kw">rbind</span>(<span class="kw">diag</span>(<span class="dv">1</span>, <span class="dt">nrow=</span><span class="kw">nrow</span>(GRM_C), <span class="dt">ncol=</span><span class="kw">nrow</span>(GRM_C)), <span class="dv">-1</span><span class="op">*</span>Pnc)</a>
<a class="sourceLine" id="cb13-13" title="13">v2_<span class="dv">2</span>=<span class="kw">rbind</span>(<span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow=</span><span class="kw">nrow</span>(GRM_C), <span class="dt">ncol=</span><span class="kw">nrow</span>(GRM_N)), <span class="kw">diag</span>(<span class="dv">1</span>, <span class="kw">nrow</span>(GRM_N), <span class="kw">ncol</span>(GRM_N)))</a>
<a class="sourceLine" id="cb13-14" title="14">V2=<span class="kw">cbind</span>(v2_<span class="dv">1</span>, v2_<span class="dv">2</span>)</a>
<a class="sourceLine" id="cb13-15" title="15"></a>
<a class="sourceLine" id="cb13-16" title="16">IV=V1 <span class="op">%*%</span><span class="st"> </span>V2</a>
<a class="sourceLine" id="cb13-17" title="17"><span class="kw">print</span>(IV) <span class="co">#apy</span></a></code></pre></div>
<pre><code>##           [,1]      [,2] [,3]      [,4]     [,5]
## [1,]  66.80498  65.57041    0 -33.19502 33.19502
## [2,]  65.57041  66.80498    0 -33.19502 33.19502
## [3,]   0.00000   0.00000  100   0.00000  0.00000
## [4,] -33.19502 -33.19502    0  33.60996  0.00000
## [5,]  33.19502  33.19502    0   0.00000 33.60996</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1">IVI=<span class="kw">solve</span>(IV)</a>
<a class="sourceLine" id="cb15-2" title="2"><span class="kw">print</span>(IVI)</a></code></pre></div>
<pre><code>##               [,1]          [,2] [,3]      [,4]      [,5]
## [1,]  8.100000e-01  9.572467e-16 0.00  0.800000 -0.800000
## [2,]  1.089201e-15  8.100000e-01 0.00  0.800000 -0.800000
## [3,]  0.000000e+00  0.000000e+00 0.01  0.000000  0.000000
## [4,]  8.000000e-01  8.000000e-01 0.00  1.610000 -1.580247
## [5,] -8.000000e-01 -8.000000e-01 0.00 -1.580247  1.610000</code></pre>
</div>
</div>
<div id="ld-score-regression" class="section level2">
<h2><span class="header-section-number">8.2</span> LD score regression</h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" title="1"><span class="kw">source</span>(<span class="st">&quot;~/R/MyLib/shotgun.R&quot;</span>)</a>
<a class="sourceLine" id="cb17-2" title="2">REP =<span class="st"> </span><span class="dv">50</span></a>
<a class="sourceLine" id="cb17-3" title="3">N =<span class="st"> </span><span class="dv">500</span></a>
<a class="sourceLine" id="cb17-4" title="4">M =<span class="st"> </span><span class="dv">1000</span></a>
<a class="sourceLine" id="cb17-5" title="5">h2 =<span class="st"> </span><span class="fl">0.5</span></a>
<a class="sourceLine" id="cb17-6" title="6"></a>
<a class="sourceLine" id="cb17-7" title="7">h2E =<span class="st"> </span><span class="kw">array</span>(<span class="dv">0</span>, <span class="dt">dim =</span> <span class="kw">c</span>(<span class="dv">2</span>, REP))</a>
<a class="sourceLine" id="cb17-8" title="8"></a>
<a class="sourceLine" id="cb17-9" title="9"><span class="cf">for</span> (rep <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>REP) {</a>
<a class="sourceLine" id="cb17-10" title="10">    fq =<span class="st"> </span><span class="kw">runif</span>(M, <span class="fl">0.05</span>, <span class="fl">0.95</span>)</a>
<a class="sourceLine" id="cb17-11" title="11">    Dl =<span class="st"> </span><span class="kw">array</span>(<span class="dv">0</span>, <span class="dt">dim =</span> M)</a>
<a class="sourceLine" id="cb17-12" title="12">    Dl =<span class="st"> </span><span class="kw">runif</span>(M, <span class="fl">0.8</span>, <span class="fl">0.9</span>)</a>
<a class="sourceLine" id="cb17-13" title="13">    Dl[<span class="kw">seq</span>(<span class="dv">10</span>, M, <span class="dv">10</span>)] =<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb17-14" title="14">    </a>
<a class="sourceLine" id="cb17-15" title="15">    G =<span class="st"> </span><span class="kw">GenerateGenoDprime</span>(fq, Dl[<span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(Dl) <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)], N)</a>
<a class="sourceLine" id="cb17-16" title="16">    b =<span class="st"> </span><span class="kw">rnorm</span>(M, <span class="dv">0</span>, <span class="kw">sqrt</span>(h2<span class="op">/</span>M))</a>
<a class="sourceLine" id="cb17-17" title="17">    bv =<span class="st"> </span>G <span class="op">%*%</span><span class="st"> </span>b</a>
<a class="sourceLine" id="cb17-18" title="18">    Y =<span class="st"> </span>bv <span class="op">+</span><span class="st"> </span><span class="kw">rnorm</span>(N, <span class="dv">0</span>, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb17-19" title="19">    chi =<span class="st"> </span><span class="kw">array</span>(<span class="dv">0</span>, <span class="dt">dim =</span> M)</a>
<a class="sourceLine" id="cb17-20" title="20">    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>M) {</a>
<a class="sourceLine" id="cb17-21" title="21">        sm =<span class="st"> </span><span class="kw">summary</span>(<span class="kw">lm</span>(Y <span class="op">~</span><span class="st"> </span>G[, i]))</a>
<a class="sourceLine" id="cb17-22" title="22">        chi[i] =<span class="st"> </span>sm<span class="op">$</span>coefficients[<span class="dv">2</span>, <span class="dv">3</span>]<span class="op">^</span><span class="dv">2</span></a>
<a class="sourceLine" id="cb17-23" title="23">    }</a>
<a class="sourceLine" id="cb17-24" title="24">    <span class="kw">plot</span>(chi, b<span class="op">^</span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>fq <span class="op">*</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>fq), <span class="dt">pch =</span> <span class="dv">16</span>)</a>
<a class="sourceLine" id="cb17-25" title="25">    lds =<span class="st"> </span><span class="kw">array</span>(<span class="dv">0</span>, <span class="dt">dim =</span> M)</a>
<a class="sourceLine" id="cb17-26" title="26">    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(M<span class="op">/</span><span class="dv">10</span>)) {</a>
<a class="sourceLine" id="cb17-27" title="27">        cg =<span class="st"> </span><span class="kw">cor</span>(G[, ((i <span class="op">-</span><span class="st"> </span><span class="dv">1</span>) <span class="op">*</span><span class="st"> </span><span class="dv">10</span> <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)<span class="op">:</span>(i <span class="op">*</span><span class="st"> </span><span class="dv">10</span>)])</a>
<a class="sourceLine" id="cb17-28" title="28">        <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>) {</a>
<a class="sourceLine" id="cb17-29" title="29">            lds[(i <span class="op">-</span><span class="st"> </span><span class="dv">1</span>) <span class="op">*</span><span class="st"> </span><span class="dv">10</span> <span class="op">+</span><span class="st"> </span>j] =<span class="st"> </span><span class="kw">sum</span>(cg[, j]<span class="op">^</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb17-30" title="30">        }</a>
<a class="sourceLine" id="cb17-31" title="31">    }</a>
<a class="sourceLine" id="cb17-32" title="32">    ldSc =<span class="st"> </span><span class="kw">lm</span>(chi <span class="op">~</span><span class="st"> </span>lds)</a>
<a class="sourceLine" id="cb17-33" title="33">    h2E[<span class="dv">1</span>, rep] =<span class="st"> </span>ldSc<span class="op">$</span>coefficients[<span class="dv">2</span>] <span class="op">*</span><span class="st"> </span>M<span class="op">/</span>N</a>
<a class="sourceLine" id="cb17-34" title="34">    h2E[<span class="dv">2</span>, rep] =<span class="st"> </span><span class="kw">var</span>(bv)<span class="op">/</span><span class="kw">var</span>(Y)</a>
<a class="sourceLine" id="cb17-35" title="35">}</a>
<a class="sourceLine" id="cb17-36" title="36"><span class="kw">plot</span>(h2E[<span class="dv">1</span>, ], h2E[<span class="dv">2</span>, ])</a></code></pre></div>
</div>
<div id="matrix-inversion-wiki" class="section level2">
<h2><span class="header-section-number">8.3</span> Matrix inversion <a href="https://en.wikipedia.org/wiki/Invertible_matrix#Inversion_of_2_%C3%97_2_matrices">wiki</a></h2>
<div id="x-2" class="section level3">
<h3><span class="header-section-number">8.3.1</span> 2 X 2</h3>
<p>The cofactor equation listed above yields the following result for 2 ?? 2 matrices. Inversion of these matrices can be done as follows</p>
<p><span class="math display">\[\text{A}^{-1}=\left[ \begin{matrix} 
a &amp; b\\
c &amp; d
\end{matrix} \right] ^{-1}
=\frac{1}{det (\text{A})}\left[ \begin{matrix} 
A &amp; B\\
C &amp; D
\end{matrix} \right]
=\frac{1}{ad-bc}\left[ \begin{matrix} 
d &amp; -b\\
-c &amp; a
\end{matrix} \right]
\]</span>
Where the scalar <span class="math inline">\(A\)</span> is not to be confused with the matrix A.</p>
</div>
<div id="x-3" class="section level3">
<h3><span class="header-section-number">8.3.2</span> 3 X 3</h3>
<p><span class="math display">\[\text{A}^{-1}=\left[ \begin{matrix} 
a &amp; b &amp; c\\
d &amp; e &amp; f\\
g &amp; h &amp; i
\end{matrix} \right] ^{-1}
=\frac{1}{det (\text{A})}\left[ \begin{matrix} 
A &amp; B &amp; C\\
D &amp; E &amp; F\\
G &amp; H &amp; I
\end{matrix} \right] ^{T}
=\frac{1}{det (\text{A})}\left[ \begin{matrix} 
A &amp; D &amp; G\\
B &amp; E &amp; H\\
C &amp; F &amp; I
\end{matrix} \right]
\]</span></p>
<p>in which
<span class="math display">\[\begin{matrix}
A=(ei-fh), &amp; D=-(bi-ch), &amp; G = -(bf-ce), \\
B=-(di-fg), &amp; E=(ai-cg), &amp; H = -(af-cd), \\
C=(dh-eg), &amp; F=-(ah-bg), &amp; I = (ae-bd).
\end{matrix}
\]</span></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="references.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="IBD-fam.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["EigenGWAS.pdf", "EigenGWAS.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
