<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>EigenGWAS theory and application</title>
  <meta name="description" content="EigenGWAS Project.">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="EigenGWAS theory and application" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="EigenGWAS Project." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="EigenGWAS theory and application" />
  
  <meta name="twitter:description" content="EigenGWAS Project." />
  

<meta name="author" content="Guo-Bo Chen [chen.guobo@foxmail.com]">


<meta name="date" content="2019-05-13">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="resequencing-studies.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">EigenGWAS core</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> EigenGWAS basis</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#genetic-relatedness-matrix-mathbfg"><i class="fa fa-check"></i><b>1.1</b> Genetic relatedness matrix <span class="math inline">\(\mathbf{G}\)</span></a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#statistical-properties-of-mathbfg"><i class="fa fa-check"></i><b>1.1.1</b> Statistical properties of <span class="math inline">\(\mathbf{G}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#eigengwas-linear-model"><i class="fa fa-check"></i><b>1.2</b> EigenGWAS linear model</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#lambda_gc-correction"><i class="fa fa-check"></i><b>1.2.1</b> <span class="math inline">\(\lambda_{GC}\)</span> correction</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#threshold-for-eigengwas"><i class="fa fa-check"></i><b>1.2.2</b> Threshold for EigenGWAS</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#connection-to-singular-value-decomposition"><i class="fa fa-check"></i><b>1.3</b> Connection to singular value decomposition</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#intepretation"><i class="fa fa-check"></i><b>1.4</b> Intepretation</a><ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#f_st"><i class="fa fa-check"></i><b>1.4.1</b> <span class="math inline">\(F_{st}\)</span></a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#classic-mechanic-intepretation"><i class="fa fa-check"></i><b>1.4.2</b> Classic mechanic intepretation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="protocols.html"><a href="protocols.html"><i class="fa fa-check"></i><b>2</b> Protocols</a><ul>
<li class="chapter" data-level="2.1" data-path="protocols.html"><a href="protocols.html#protocols-for-selection"><i class="fa fa-check"></i><b>2.1</b> Protocols for selection</a></li>
<li class="chapter" data-level="2.2" data-path="protocols.html"><a href="protocols.html#protocal-for-eigengwas"><i class="fa fa-check"></i><b>2.2</b> Protocal for EigenGWAS</a></li>
<li class="chapter" data-level="2.3" data-path="protocols.html"><a href="protocols.html#protocol-for-predicted-eigenvectors"><i class="fa fa-check"></i><b>2.3</b> Protocol for predicted eigenvectors</a></li>
<li class="chapter" data-level="2.4" data-path="protocols.html"><a href="protocols.html#other-test-statistics-for-selection"><i class="fa fa-check"></i><b>2.4</b> Other test statistics for selection</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="resequencing-studies.html"><a href="resequencing-studies.html"><i class="fa fa-check"></i><b>3</b> Resequencing studies</a><ul>
<li class="chapter" data-level="3.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#technical-review"><i class="fa fa-check"></i><b>3.1</b> Technical review</a><ul>
<li class="chapter" data-level="3.1.1" data-path="resequencing-studies.html"><a href="resequencing-studies.html#ngs"><i class="fa fa-check"></i><b>3.1.1</b> NGS</a></li>
<li class="chapter" data-level="3.1.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#chip-data"><i class="fa fa-check"></i><b>3.1.2</b> Chip data</a></li>
<li class="chapter" data-level="3.1.3" data-path="resequencing-studies.html"><a href="resequencing-studies.html#gbs"><i class="fa fa-check"></i><b>3.1.3</b> GBS</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="resequencing-studies.html"><a href="resequencing-studies.html#drop-hwe-test"><i class="fa fa-check"></i><b>3.2</b> Drop HWE test</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html"><i class="fa fa-check"></i><b>4</b> Simulating population structure</a><ul>
<li class="chapter" data-level="4.1" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#genetic-drift"><i class="fa fa-check"></i><b>4.1</b> Genetic drift</a></li>
<li class="chapter" data-level="4.2" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#discrete-populations"><i class="fa fa-check"></i><b>4.2</b> Discrete populations</a></li>
<li class="chapter" data-level="4.3" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#admixture-populations"><i class="fa fa-check"></i><b>4.3</b> Admixture populations</a></li>
<li class="chapter" data-level="4.4" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#homo-heteogeneous-f_st"><i class="fa fa-check"></i><b>4.4</b> Homo &amp; Heteogeneous <span class="math inline">\(F_{st}\)</span></a></li>
<li class="chapter" data-level="4.5" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#wishart-distribution"><i class="fa fa-check"></i><b>4.5</b> Wishart distribution</a></li>
<li class="chapter" data-level="4.6" data-path="simulating-population-structure.html"><a href="simulating-population-structure.html#tracy-widom-distribution"><i class="fa fa-check"></i><b>4.6</b> Tracy-Widom distribution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>5</b> Data analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="data-analysis.html"><a href="data-analysis.html#on-site-examples"><i class="fa fa-check"></i><b>5.1</b> On-site examples</a><ul>
<li class="chapter" data-level="5.1.1" data-path="data-analysis.html"><a href="data-analysis.html#arabdiopsis"><i class="fa fa-check"></i><b>5.1.1</b> Arabdiopsis</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-analysis.html"><a href="data-analysis.html#k-rice"><i class="fa fa-check"></i><b>5.1.2</b> 3K rice</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-analysis.html"><a href="data-analysis.html#als"><i class="fa fa-check"></i><b>5.1.3</b> ALS</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-analysis.html"><a href="data-analysis.html#gf"><i class="fa fa-check"></i><b>5.1.4</b> GF</a></li>
<li class="chapter" data-level="5.1.5" data-path="data-analysis.html"><a href="data-analysis.html#uk-birds"><i class="fa fa-check"></i><b>5.1.5</b> UK birds</a></li>
<li class="chapter" data-level="5.1.6" data-path="data-analysis.html"><a href="data-analysis.html#darwins-finches"><i class="fa fa-check"></i><b>5.1.6</b> Darwin’s finches</a></li>
<li class="chapter" data-level="5.1.7" data-path="data-analysis.html"><a href="data-analysis.html#gbs-maize"><i class="fa fa-check"></i><b>5.1.7</b> GBS Maize</a></li>
<li class="chapter" data-level="5.1.8" data-path="data-analysis.html"><a href="data-analysis.html#uk-biobank"><i class="fa fa-check"></i><b>5.1.8</b> UK Biobank</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-analysis.html"><a href="data-analysis.html#public-datahub-neo"><i class="fa fa-check"></i><b>5.2</b> Public datahub (NEO)</a></li>
<li class="chapter" data-level="5.3" data-path="data-analysis.html"><a href="data-analysis.html#meta-scale"><i class="fa fa-check"></i><b>5.3</b> Meta-scale</a><ul>
<li class="chapter" data-level="5.3.1" data-path="data-analysis.html"><a href="data-analysis.html#n_e"><i class="fa fa-check"></i><b>5.3.1</b> <span class="math inline">\(n_e\)</span></a></li>
<li class="chapter" data-level="5.3.2" data-path="data-analysis.html"><a href="data-analysis.html#m_e"><i class="fa fa-check"></i><b>5.3.2</b> <span class="math inline">\(m_e\)</span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a><ul>
<li class="chapter" data-level="6.1" data-path="conclusion.html"><a href="conclusion.html#statistical-power"><i class="fa fa-check"></i><b>6.1</b> Statistical power</a></li>
<li class="chapter" data-level="6.2" data-path="conclusion.html"><a href="conclusion.html#selection-pattern"><i class="fa fa-check"></i><b>6.2</b> Selection pattern</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="app-1.html"><a href="app-1.html"><i class="fa fa-check"></i><b>7</b> Appendix 1</a><ul>
<li class="chapter" data-level="7.1" data-path="app-1.html"><a href="app-1.html#app:ld"><i class="fa fa-check"></i><b>7.1</b> Notes on linkage disequilibrium</a></li>
<li class="chapter" data-level="7.2" data-path="app-1.html"><a href="app-1.html#the-maximal-value-rho_ab"><i class="fa fa-check"></i><b>7.2</b> The maximal value <span class="math inline">\(\rho_{AB}\)</span></a></li>
<li class="chapter" data-level="7.3" data-path="app-1.html"><a href="app-1.html#ld-and-recombination-fraction"><i class="fa fa-check"></i><b>7.3</b> LD and recombination fraction</a></li>
<li class="chapter" data-level="7.4" data-path="app-1.html"><a href="app-1.html#f_st-1"><i class="fa fa-check"></i><b>7.4</b> <span class="math inline">\(F_{st}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://gc5k.github.io" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">EigenGWAS theory and application</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="protocols" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Protocols</h1>
<div id="protocols-for-selection" class="section level2">
<h2><span class="header-section-number">2.1</span> Protocols for selection</h2>
<ol style="list-style-type: decimal">
<li><p>Constructing genetic relationship matrix <span class="math inline">\(\mathbf{G}\)</span>. The difference between a random mating population and inbred population is the way <span class="math inline">\(\mathbf{G}\)</span> is constructed. So for a pair of individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>,
<span class="math display">\[G_{ij}=\frac{1}{\tilde{m}}\sum_l^{\tilde{m}}\frac{(x_{il}-2p_l)(x_{jl}-2p_l)}{2(1+F)p_lq_l}\]</span>
and <span class="math inline">\(F=1\)</span> for inbred lines. Please refer to Chapter <a href="index.html#ch:basis">1</a>.</p></li>
<li><p>Conducting eigenanalysis for <span class="math inline">\(\mathbf{G}\)</span>, see it definition <a href="#eq:G">(<strong>??</strong>)</a>.</p></li>
<li><p>Linear regression analysis for each SNP.</p></li>
</ol>
</div>
<div id="protocal-for-eigengwas" class="section level2">
<h2><span class="header-section-number">2.2</span> Protocal for EigenGWAS</h2>
<p>A simple analysis pipeline written in Rscripts may be found at <a href="http://rpubs.com/gc5k/EigenGWASpipeline">Rpub</a> and its
<a href="https://gc5k.shinyapps.io/eigengwas/">Shiny demo</a>.</p>
<p>Or, try it as below in <code>RStudio</code></p>
<p><code>runGitHub(&quot;Rshiny&quot;, &quot;gc5k&quot;)</code></p>
</div>
<div id="protocol-for-predicted-eigenvectors" class="section level2">
<h2><span class="header-section-number">2.3</span> Protocol for predicted eigenvectors</h2>
<p>The prediction accuracy can be written as
<span class="math display">\[R^2 \approx \frac{1}{1+\frac{n_e}{m}}\]</span>
## Nucleotide diversity</p>
<p><span class="math inline">\(\theta\)</span>, the normalized observed number of variants sites</p>
<p><span class="math inline">\(\pi\)</span>, the observed heterozygosity per base pair</p>
<blockquote>
<p>Nonetheless, we can expect that favored alleles will generally sit whthin large shared haplotypes, and that these haplotypes will be in sharp contrast with more variable haplotypes on the unselected background. Voight, et al, <strong>PLoS Biology</strong>, 2006, 4, e72.</p>
</blockquote>
<blockquote>
<p>However, the excess variability around the site of F/S amino-acid polymorphism is found within S haplotypes, and F haplotypes are depauperate in variability, suggesting that the F allele is a derived variant that has recently swept to an intermediate frequency. Charsworth &amp; Charsworth, <strong>Heredity</strong>, 2017, 118:2-9.</p>
</blockquote>
</div>
<div id="other-test-statistics-for-selection" class="section level2">
<h2><span class="header-section-number">2.4</span> Other test statistics for selection</h2>
<p><strong>1 Tajima’s</strong> <span class="math inline">\(D\)</span>, Statistical method for testing the neutral mutation hypothesis by DNA polymorphism, 1989, <span class="math inline">\(Genetics\)</span>, 123:585-95</p>
<p><strong>2 Fu &amp; Li’s</strong> <span class="math inline">\(D\)</span>, Statistical tests of neutrality of mutations, 1993, <span class="math inline">\(Genetics\)</span>, 133:693-709</p>
<p><strong>3 Fay &amp; Wu’s</strong> <span class="math inline">\(H\)</span>, Hitchhiking under positive Darwinian selection, 2000, <span class="math inline">\(Genetics\)</span>, 155:1405-13</p>
<p><strong>4 McDonald and Kreitman test</strong>, Adaptive protein evolution at the Adh locus in Drosophila, 1991, <span class="math inline">\(Nature\)</span>, 335:167-70</p>
<p><strong>5 HKA test</strong>, Hudson, Kreitman, Aguade, A test of neutral molecular evolution based on nucleotide data, 1987, <span class="math inline">\(Genetics\)</span>, 166:153-9)</p>
<p><strong>6 Extended haplotype homozygosity (EHH)</strong>
See Voight et al <span class="math inline">\(PLoS Biology, 2006, 4, e72\)</span></p>
<blockquote>
<p>Unstandardized
<span class="math inline">\(iHS=ln\frac{iHH_A}{iHH_D}\)</span></p>
</blockquote>
<blockquote>
<p>and standardized
<span class="math inline">\(iHS=\frac{ln[\frac{iHH_A}{iHH_D}]-E_p[ln(\frac{iHH_A}{iHH_D})]}{SD_P[ln(\frac{iHH_A}{iHH_D})]}\)</span></p>
</blockquote>
<p>rehh: an R package to detect footprints of selection in genome-wide SNP data from haplotype structure, <span class="math inline">\(Bioinformatics\)</span>, 2012, 8:1176-7</p>
<p>selscan: An Efficient Multithreaded Program to Perform EHH-Based Scans for Positive Selection, 2014, <span class="math inline">\(MBE\)</span>, 31:2824-7, <a href="https://github.com/szpiech/selscan">selcan’s github</a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="resequencing-studies.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["EigenGWAS.pdf", "EigenGWAS.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
